{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "BEE_TEAM_QIIME2_SRA.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "toc_visible": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "j2yT4h7_PooP",
        "outputId": "3852693b-3101-4883-ca4a-c318d86e5dd1"
      },
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "hh3TMIgxPoom"
      },
      "source": [
        "!cp /content/sample_data/study1/*.gz /content/drive/MyDrive/BeeTeam/studyx/ "
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ReImKqjCyzQt"
      },
      "source": [
        "#Start here: Add google drive, files and install conda"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "70ZkHUX2l_Qh"
      },
      "source": [
        "conda install parallel-fastq-dump"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "M9o8Y_B7MgFl",
        "outputId": "0725a357-f101-4643-f7ae-3ba6e5d70b13"
      },
      "source": [
        "!wget -c https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n",
        "!chmod +x Miniconda3-latest-Linux-x86_64.sh\n",
        "!bash ./Miniconda3-latest-Linux-x86_64.sh -b -f -p /usr/local\n",
        "# can change to another channel or package\n",
        "#!conda config --add channels defaults\n",
        "#!conda config --add channels defaults\n",
        "#!conda config --add channels bioconda\n",
        "#!conda config --add channels conda-forge\n",
        "#!conda install parallel-fastq-dump\n",
        "import sys\n",
        "sys.path.append(\"/usr/local/conda/lib/python3.6/site-packages/\")"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "--2021-03-21 19:35:12--  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n",
            "Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.130.3, 104.16.131.3, 2606:4700::6810:8203, ...\n",
            "Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.130.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 94235922 (90M) [application/x-sh]\n",
            "Saving to: ‘Miniconda3-latest-Linux-x86_64.sh’\n",
            "\n",
            "Miniconda3-latest-L 100%[===================>]  89.87M   162MB/s    in 0.6s    \n",
            "\n",
            "2021-03-21 19:35:12 (162 MB/s) - ‘Miniconda3-latest-Linux-x86_64.sh’ saved [94235922/94235922]\n",
            "\n",
            "PREFIX=/usr/local\n",
            "Unpacking payload ...\n",
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - _libgcc_mutex==0.1=main\n",
            "    - brotlipy==0.7.0=py38h27cfd23_1003\n",
            "    - ca-certificates==2020.10.14=0\n",
            "    - certifi==2020.6.20=pyhd3eb1b0_3\n",
            "    - cffi==1.14.3=py38h261ae71_2\n",
            "    - chardet==3.0.4=py38h06a4308_1003\n",
            "    - conda-package-handling==1.7.2=py38h03888b9_0\n",
            "    - conda==4.9.2=py38h06a4308_0\n",
            "    - cryptography==3.2.1=py38h3c74f83_1\n",
            "    - idna==2.10=py_0\n",
            "    - ld_impl_linux-64==2.33.1=h53a641e_7\n",
            "    - libedit==3.1.20191231=h14c3975_1\n",
            "    - libffi==3.3=he6710b0_2\n",
            "    - libgcc-ng==9.1.0=hdf63c60_0\n",
            "    - libstdcxx-ng==9.1.0=hdf63c60_0\n",
            "    - ncurses==6.2=he6710b0_1\n",
            "    - openssl==1.1.1h=h7b6447c_0\n",
            "    - pip==20.2.4=py38h06a4308_0\n",
            "    - pycosat==0.6.3=py38h7b6447c_1\n",
            "    - pycparser==2.20=py_2\n",
            "    - pyopenssl==19.1.0=pyhd3eb1b0_1\n",
            "    - pysocks==1.7.1=py38h06a4308_0\n",
            "    - python==3.8.5=h7579374_1\n",
            "    - readline==8.0=h7b6447c_0\n",
            "    - requests==2.24.0=py_0\n",
            "    - ruamel_yaml==0.15.87=py38h7b6447c_1\n",
            "    - setuptools==50.3.1=py38h06a4308_1\n",
            "    - six==1.15.0=py38h06a4308_0\n",
            "    - sqlite==3.33.0=h62c20be_0\n",
            "    - tk==8.6.10=hbc83047_0\n",
            "    - tqdm==4.51.0=pyhd3eb1b0_0\n",
            "    - urllib3==1.25.11=py_0\n",
            "    - wheel==0.35.1=pyhd3eb1b0_0\n",
            "    - xz==5.2.5=h7b6447c_0\n",
            "    - yaml==0.2.5=h7b6447c_0\n",
            "    - zlib==1.2.11=h7b6447c_3\n",
            "\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main\n",
            "  brotlipy           pkgs/main/linux-64::brotlipy-0.7.0-py38h27cfd23_1003\n",
            "  ca-certificates    pkgs/main/linux-64::ca-certificates-2020.10.14-0\n",
            "  certifi            pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3\n",
            "  cffi               pkgs/main/linux-64::cffi-1.14.3-py38h261ae71_2\n",
            "  chardet            pkgs/main/linux-64::chardet-3.0.4-py38h06a4308_1003\n",
            "  conda              pkgs/main/linux-64::conda-4.9.2-py38h06a4308_0\n",
            "  conda-package-han~ pkgs/main/linux-64::conda-package-handling-1.7.2-py38h03888b9_0\n",
            "  cryptography       pkgs/main/linux-64::cryptography-3.2.1-py38h3c74f83_1\n",
            "  idna               pkgs/main/noarch::idna-2.10-py_0\n",
            "  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7\n",
            "  libedit            pkgs/main/linux-64::libedit-3.1.20191231-h14c3975_1\n",
            "  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2\n",
            "  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0\n",
            "  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0\n",
            "  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1\n",
            "  openssl            pkgs/main/linux-64::openssl-1.1.1h-h7b6447c_0\n",
            "  pip                pkgs/main/linux-64::pip-20.2.4-py38h06a4308_0\n",
            "  pycosat            pkgs/main/linux-64::pycosat-0.6.3-py38h7b6447c_1\n",
            "  pycparser          pkgs/main/noarch::pycparser-2.20-py_2\n",
            "  pyopenssl          pkgs/main/noarch::pyopenssl-19.1.0-pyhd3eb1b0_1\n",
            "  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py38h06a4308_0\n",
            "  python             pkgs/main/linux-64::python-3.8.5-h7579374_1\n",
            "  readline           pkgs/main/linux-64::readline-8.0-h7b6447c_0\n",
            "  requests           pkgs/main/noarch::requests-2.24.0-py_0\n",
            "  ruamel_yaml        pkgs/main/linux-64::ruamel_yaml-0.15.87-py38h7b6447c_1\n",
            "  setuptools         pkgs/main/linux-64::setuptools-50.3.1-py38h06a4308_1\n",
            "  six                pkgs/main/linux-64::six-1.15.0-py38h06a4308_0\n",
            "  sqlite             pkgs/main/linux-64::sqlite-3.33.0-h62c20be_0\n",
            "  tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0\n",
            "  tqdm               pkgs/main/noarch::tqdm-4.51.0-pyhd3eb1b0_0\n",
            "  urllib3            pkgs/main/noarch::urllib3-1.25.11-py_0\n",
            "  wheel              pkgs/main/noarch::wheel-0.35.1-pyhd3eb1b0_0\n",
            "  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0\n",
            "  yaml               pkgs/main/linux-64::yaml-0.2.5-h7b6447c_0\n",
            "  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3\n",
            "\n",
            "\n",
            "Preparing transaction: / \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "installation finished.\n",
            "WARNING:\n",
            "    You currently have a PYTHONPATH environment variable set. This may cause\n",
            "    unexpected behavior when running the Python interpreter in Miniconda3.\n",
            "    For best results, please verify that your PYTHONPATH only points to\n",
            "    directories of packages that are compatible with the Python interpreter\n",
            "    in Miniconda3: /usr/local\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5S8nIaWhy4Iv"
      },
      "source": [
        "#**Study 2** "
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LVj3v-hQvuqr",
        "outputId": "e95cf996-baac-4827-c730-723dca730005"
      },
      "source": [
        "!wget -r \"https://datadryad.org/stash/downloads/file_stream/514406\""
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "--2021-02-24 11:15:15--  https://datadryad.org/stash/downloads/file_stream/514406\n",
            "Resolving datadryad.org (datadryad.org)... 44.235.214.65, 44.239.58.150, 52.43.155.29\n",
            "Connecting to datadryad.org (datadryad.org)|44.235.214.65|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://dryad-assetstore-merritt-west.s3.us-west-2.amazonaws.com/ark%3A/13030/m5tn2h5k%7C1%7Cproducer/FastqFiles_MS0162_SimoneFinstrom_16Spool.tar?response-content-type=application%2Fx-tar&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEPv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIAEaU0pC4hrSpaGLxH4L1dO28n882lg8TUGCJkGr3m1qAiAbFGyRjOJmJ6GoC1aq5%2BDoZV5XD9rLQqg9xGsdeMuP0yq0AwgUEAAaDDQ1MTgyNjkxNDE1NyIMwcVYLK0uIA71H7eiKpEDxZ%2FaWPSQ1EhHZz7IYE2wlVgyaNCY8JbVaX6T7XwjJn47onveoVSB4AqzZJHw5YhWYaJ%2FRx6wVjzHiLhETMn2lrE64mISumyhKGAIFwLpkQ1zDTlhGL5wX3sFXeYQ55AX2fROSZ9hEzcWhroLgzonqi5SivCJPC4cPQCwygYjKV5AEsF7vNFjm0jJqqElpZ6wigpV93VchNaVHE%2Bv2dbPWlSfPTEC4Zy0rd81gG1YMbxzbMOCtH0MKIH%2FQXzH5luPkUFdCr%2B6%2Bqm%2FdUoy6%2Fkm1lxlI39a2QqOWQnHzttwLiedYPcltEbn3Dt7%2B%2BRJbC3vabOuE3PX2XxOjE2fCEHTi%2F7X7Uzvk7cV4b9L8q8VBI9UcfNw367VZnCpwBwltgX3OjepSv%2FCSzsWohQogzTgJuh%2FG%2BrKXSZRqfz8U2p0osaB0NiHjPD8YAg8DxjF5NuHsXTepPoDIAr80Gr6o8RbRuIwyaQcTN0ms4%2FNk6bm4sO5yYU9tIBZxExTlf9tAETixzjywsYj8%2BbNRP4IJtWFCEcw9NHYgQY67AGH5E8um1NB6lWp6%2Fb%2FTnjLW3%2BV2njOomLdISD%2BhQLt%2FJO8ltCzoLXikgTB19JfSkNnaU1l0qZkaAJqacJdWzCNNsaNiSZmDd3Eijl0Ryaz9BR1z09nieA3tsUY0NyPnIA3kxEKkw4Ro%2F6p5HtO7qs2nUCq%2BdNfKAw5LylG1a36ywTqDncSr8KlRjkhoWVC2Oo2Mdr%2FQ%2Br0ssdGch0J1%2FXCBD%2FRUeooBPDQ59qaoZyjfdecRQ8SBcicYjW6xTamx8xCACBTPxXfP3sHKmQjo0kzs4ylTNu6vMGN%2B6DQEuUO2j%2BFIC0xI3FfDeFY7w%3D%3D&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210224T111516Z&X-Amz-SignedHeaders=host&X-Amz-Expires=14400&X-Amz-Credential=ASIAWSMX3SNWTCKQ7ZYG%2F20210224%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Signature=e700d89089387f92356870648804652fed05ccf8b956bb52a43160446bfeb014 [following]\n",
            "--2021-02-24 11:15:16--  https://dryad-assetstore-merritt-west.s3.us-west-2.amazonaws.com/ark%3A/13030/m5tn2h5k%7C1%7Cproducer/FastqFiles_MS0162_SimoneFinstrom_16Spool.tar?response-content-type=application%2Fx-tar&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEPv%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIAEaU0pC4hrSpaGLxH4L1dO28n882lg8TUGCJkGr3m1qAiAbFGyRjOJmJ6GoC1aq5%2BDoZV5XD9rLQqg9xGsdeMuP0yq0AwgUEAAaDDQ1MTgyNjkxNDE1NyIMwcVYLK0uIA71H7eiKpEDxZ%2FaWPSQ1EhHZz7IYE2wlVgyaNCY8JbVaX6T7XwjJn47onveoVSB4AqzZJHw5YhWYaJ%2FRx6wVjzHiLhETMn2lrE64mISumyhKGAIFwLpkQ1zDTlhGL5wX3sFXeYQ55AX2fROSZ9hEzcWhroLgzonqi5SivCJPC4cPQCwygYjKV5AEsF7vNFjm0jJqqElpZ6wigpV93VchNaVHE%2Bv2dbPWlSfPTEC4Zy0rd81gG1YMbxzbMOCtH0MKIH%2FQXzH5luPkUFdCr%2B6%2Bqm%2FdUoy6%2Fkm1lxlI39a2QqOWQnHzttwLiedYPcltEbn3Dt7%2B%2BRJbC3vabOuE3PX2XxOjE2fCEHTi%2F7X7Uzvk7cV4b9L8q8VBI9UcfNw367VZnCpwBwltgX3OjepSv%2FCSzsWohQogzTgJuh%2FG%2BrKXSZRqfz8U2p0osaB0NiHjPD8YAg8DxjF5NuHsXTepPoDIAr80Gr6o8RbRuIwyaQcTN0ms4%2FNk6bm4sO5yYU9tIBZxExTlf9tAETixzjywsYj8%2BbNRP4IJtWFCEcw9NHYgQY67AGH5E8um1NB6lWp6%2Fb%2FTnjLW3%2BV2njOomLdISD%2BhQLt%2FJO8ltCzoLXikgTB19JfSkNnaU1l0qZkaAJqacJdWzCNNsaNiSZmDd3Eijl0Ryaz9BR1z09nieA3tsUY0NyPnIA3kxEKkw4Ro%2F6p5HtO7qs2nUCq%2BdNfKAw5LylG1a36ywTqDncSr8KlRjkhoWVC2Oo2Mdr%2FQ%2Br0ssdGch0J1%2FXCBD%2FRUeooBPDQ59qaoZyjfdecRQ8SBcicYjW6xTamx8xCACBTPxXfP3sHKmQjo0kzs4ylTNu6vMGN%2B6DQEuUO2j%2BFIC0xI3FfDeFY7w%3D%3D&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210224T111516Z&X-Amz-SignedHeaders=host&X-Amz-Expires=14400&X-Amz-Credential=ASIAWSMX3SNWTCKQ7ZYG%2F20210224%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Signature=e700d89089387f92356870648804652fed05ccf8b956bb52a43160446bfeb014\n",
            "Resolving dryad-assetstore-merritt-west.s3.us-west-2.amazonaws.com (dryad-assetstore-merritt-west.s3.us-west-2.amazonaws.com)... 52.218.218.209\n",
            "Connecting to dryad-assetstore-merritt-west.s3.us-west-2.amazonaws.com (dryad-assetstore-merritt-west.s3.us-west-2.amazonaws.com)|52.218.218.209|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 7980523520 (7.4G) [application/x-tar]\n",
            "Saving to: ‘datadryad.org/stash/downloads/file_stream/514406’\n",
            "\n",
            "datadryad.org/stash 100%[===================>]   7.43G  18.3MB/s    in 7m 22s  \n",
            "\n",
            "2021-02-24 11:22:39 (17.2 MB/s) - ‘datadryad.org/stash/downloads/file_stream/514406’ saved [7980523520/7980523520]\n",
            "\n",
            "FINISHED --2021-02-24 11:22:39--\n",
            "Total wall clock time: 7m 24s\n",
            "Downloaded: 1 files, 7.4G in 7m 22s (17.2 MB/s)\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "XLpLoase1Beu"
      },
      "source": [
        "!cp /content/sample_data/study2/FastqFiles_MS0162_SimoneFinstrom_16Spool/*.* /content/drive/MyDrive/BeeTeam/study2/"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2Y5O_y9rdxW7",
        "outputId": "d45c43e3-dfcd-4c94-f329-ad046b76c14e"
      },
      "source": [
        "!tar -xvf /content/sample_data/study2/514406"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A1_S1_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A1_S1_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A2_S2_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A2_S2_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A3_S3_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A3_S3_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A4_S4_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A4_S4_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A5_S5_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A5_S5_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A6_S6_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A6_S6_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A7_S7_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A7_S7_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A8_S8_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A8_S8_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A9_S9_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/A9_S9_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B1_S10_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B1_S10_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B2_S11_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B2_S11_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B3_S12_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B3_S12_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B4_S13_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B4_S13_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B5_S14_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B5_S14_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B6_S15_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B6_S15_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B7_S16_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B7_S16_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B8_S17_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B8_S17_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B9_S18_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/B9_S18_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C1_S19_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C1_S19_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C2_S20_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C2_S20_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C3_S21_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C3_S21_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C4_S22_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C4_S22_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C5_S23_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C5_S23_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C6_S24_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C6_S24_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C7_S25_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C7_S25_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C8_S26_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C8_S26_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C9_S27_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/C9_S27_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D1_S28_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D1_S28_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D2_S29_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D2_S29_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D3_S30_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D3_S30_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D4_S31_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D4_S31_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D5_S32_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D5_S32_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D6_S33_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D6_S33_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D7_S34_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D7_S34_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D8_S35_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D8_S35_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D9_S36_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/D9_S36_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E1_S37_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E1_S37_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E2_S38_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E2_S38_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E3_S39_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E3_S39_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E4_S40_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E4_S40_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E5_S41_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E5_S41_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E6_S42_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E6_S42_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E7_S43_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E7_S43_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E8_S44_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/E8_S44_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F1_S45_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F1_S45_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F2_S46_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F2_S46_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F3_S47_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F3_S47_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F4_S48_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F4_S48_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F5_S49_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F5_S49_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F6_S50_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F6_S50_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F7_S51_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F7_S51_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F8_S52_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/F8_S52_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G1_S53_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G1_S53_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G2_S54_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G2_S54_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G3_S55_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G3_S55_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G4_S56_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G4_S56_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G5_S57_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G5_S57_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G6_S58_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G6_S58_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G7_S59_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G7_S59_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G8_S60_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/G8_S60_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H1_S61_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H1_S61_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H2_S62_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H2_S62_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H3_S63_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H3_S63_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H4_S64_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H4_S64_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H5_S65_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H5_S65_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H6_S66_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H6_S66_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H7_S67_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H7_S67_L001_R2_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H8_S68_L001_R1_001.fastq.gz\n",
            "FastqFiles_MS0162_SimoneFinstrom_16Spool/H8_S68_L001_R2_001.fastq.gz\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "_RfxhTKdgH8J"
      },
      "source": [
        "#**Other studies from the SRA**"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "dfnYMndJyVia",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9bfb48d2-c126-41f4-b308-aa9df0ca64b0"
      },
      "source": [
        "!wget \"https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.10.8/sratoolkit.2.10.8-ubuntu64.tar.gz\""
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "--2021-03-05 12:53:45--  https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.10.8/sratoolkit.2.10.8-ubuntu64.tar.gz\n",
            "Resolving ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)... 130.14.250.10, 130.14.250.11, 2607:f220:41e:250::11, ...\n",
            "Connecting to ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)|130.14.250.10|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 97794970 (93M) [application/x-gzip]\n",
            "Saving to: ‘sratoolkit.2.10.8-ubuntu64.tar.gz’\n",
            "\n",
            "sratoolkit.2.10.8-u 100%[===================>]  93.26M   122MB/s    in 0.8s    \n",
            "\n",
            "2021-03-05 12:53:46 (122 MB/s) - ‘sratoolkit.2.10.8-ubuntu64.tar.gz’ saved [97794970/97794970]\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "6C3GU6WMyYpV",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9ab42b2b-1809-40f7-d35d-55ab0e4e1992"
      },
      "source": [
        "!tar -xvf  /content/sratoolkit.2.10.8-ubuntu64.tar.gz"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "sratoolkit.2.10.8-ubuntu64/\n",
            "sratoolkit.2.10.8-ubuntu64/example/\n",
            "sratoolkit.2.10.8-ubuntu64/example/perl/\n",
            "sratoolkit.2.10.8-ubuntu64/example/perl/base-stats.pl\n",
            "sratoolkit.2.10.8-ubuntu64/example/perl/dump-reference.pl\n",
            "sratoolkit.2.10.8-ubuntu64/example/perl/gene-lookup.pl\n",
            "sratoolkit.2.10.8-ubuntu64/example/perl/mismatch-stats.pl\n",
            "sratoolkit.2.10.8-ubuntu64/example/perl/quality-stats.pl\n",
            "sratoolkit.2.10.8-ubuntu64/example/perl/simplefastq.pl\n",
            "sratoolkit.2.10.8-ubuntu64/example/perl/splitfastq.pl\n",
            "sratoolkit.2.10.8-ubuntu64/README-blastn\n",
            "sratoolkit.2.10.8-ubuntu64/README-vdb-config\n",
            "sratoolkit.2.10.8-ubuntu64/README.md\n",
            "sratoolkit.2.10.8-ubuntu64/schema/\n",
            "sratoolkit.2.10.8-ubuntu64/schema/align/\n",
            "sratoolkit.2.10.8-ubuntu64/schema/align/align.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/align/mate-cache.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/align/pileup-stats.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/align/qstat.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/align/refseq.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/align/seq.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/csra2/\n",
            "sratoolkit.2.10.8-ubuntu64/schema/csra2/csra2.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/csra2/read.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/csra2/reference.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/csra2/stats.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/insdc/\n",
            "sratoolkit.2.10.8-ubuntu64/schema/insdc/insdc.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/insdc/seq.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/insdc/sra.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/clip.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/ncbi.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/pnbrdb.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/seq-graph.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/seq.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/spotname.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/sra.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/stats.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/trace.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/varloc.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/ncbi/wgs-contig.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/454.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/abi.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/generic-fastq.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/helicos.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/illumina.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/ion-torrent.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/nanopore.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/pacbio.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/sra/pevents.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/vdb/\n",
            "sratoolkit.2.10.8-ubuntu64/schema/vdb/built-in.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/schema/vdb/vdb.vschema\n",
            "sratoolkit.2.10.8-ubuntu64/bin/\n",
            "sratoolkit.2.10.8-ubuntu64/bin/ncbi/\n",
            "sratoolkit.2.10.8-ubuntu64/bin/ncbi/default.kfg\n",
            "sratoolkit.2.10.8-ubuntu64/bin/ncbi/certs.kfg\n",
            "sratoolkit.2.10.8-ubuntu64/bin/ncbi/vdb-copy.kfg\n",
            "sratoolkit.2.10.8-ubuntu64/bin/abi-dump\n",
            "sratoolkit.2.10.8-ubuntu64/bin/abi-dump.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/abi-dump.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/abi-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/abi-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/abi-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-tblastn\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-blastn\n",
            "sratoolkit.2.10.8-ubuntu64/bin/align-info\n",
            "sratoolkit.2.10.8-ubuntu64/bin/align-info.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/align-info.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/bam-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/bam-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/bam-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/blastn_vdb.2.10.1-2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/cache-mgr\n",
            "sratoolkit.2.10.8-ubuntu64/bin/cache-mgr.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/cache-mgr.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/cg-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/cg-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/cg-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/dump-ref-fasta\n",
            "sratoolkit.2.10.8-ubuntu64/bin/dump-ref-fasta.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/dump-ref-fasta.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fasterq-dump\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fasterq-dump-orig.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fasterq-dump.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fasterq-dump.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fastq-dump\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fastq-dump-orig.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fastq-dump.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fastq-dump.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fastq-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fastq-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/fastq-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/helicos-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/helicos-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/helicos-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/illumina-dump\n",
            "sratoolkit.2.10.8-ubuntu64/bin/illumina-dump.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/illumina-dump.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/illumina-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/illumina-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/illumina-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/kar\n",
            "sratoolkit.2.10.8-ubuntu64/bin/kar.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/kar.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/kdbmeta\n",
            "sratoolkit.2.10.8-ubuntu64/bin/kdbmeta.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/kdbmeta.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/latf-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/latf-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/latf-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/pacbio-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/pacbio-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/pacbio-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/prefetch\n",
            "sratoolkit.2.10.8-ubuntu64/bin/prefetch-orig.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/prefetch.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/prefetch.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/rcexplain\n",
            "sratoolkit.2.10.8-ubuntu64/bin/rcexplain.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/rcexplain.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/remote-fuser\n",
            "sratoolkit.2.10.8-ubuntu64/bin/remote-fuser.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/remote-fuser.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sam-dump\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sam-dump-orig.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sam-dump.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sam-dump.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sff-dump\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sff-dump.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sff-dump.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sff-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sff-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sff-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-pileup\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-pileup-orig.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-pileup.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-pileup.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-search\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-search.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-search.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-sort\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-sort-cg\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-sort-cg.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-sort-cg.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-sort.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-sort.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-stat\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-stat.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sra-stat.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/srapath\n",
            "sratoolkit.2.10.8-ubuntu64/bin/srapath-orig.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/srapath.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/srapath.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/sratools.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/srf-load\n",
            "sratoolkit.2.10.8-ubuntu64/bin/srf-load.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/srf-load.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/tblastn_vdb.2.10.1-2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/test-sra\n",
            "sratoolkit.2.10.8-ubuntu64/bin/test-sra.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/test-sra.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-config\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-config.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-config.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-copy\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-copy.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-copy.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-decrypt\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-decrypt.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-decrypt.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-dump\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-dump-orig.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-dump.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-dump.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-encrypt\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-encrypt.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-encrypt.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-lock\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-lock.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-lock.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-passwd\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-passwd.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-passwd.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-unlock\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-unlock.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-unlock.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-validate\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-validate.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/vdb-validate.2.10.8\n",
            "sratoolkit.2.10.8-ubuntu64/bin/blastn_vdb.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/blastn_vdb\n",
            "sratoolkit.2.10.8-ubuntu64/bin/tblastn_vdb.2\n",
            "sratoolkit.2.10.8-ubuntu64/bin/tblastn_vdb\n",
            "sratoolkit.2.10.8-ubuntu64/CHANGES\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "kIHjZmNByYee",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "5bba4b50-a142-409c-a0b0-c350827d43c0"
      },
      "source": [
        "cd /content/sratoolkit.2.10.8-ubuntu64/bin"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/sratoolkit.2.10.8-ubuntu64/bin\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "VJdJXQesyYQ0",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1abb9097-0978-4793-9105-d8f5312d4a86"
      },
      "source": [
        "%%bash\n",
        "./vdb-config --interactive "
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Process is terminated.\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "EoP5yB6nH9za"
      },
      "source": [
        "!echo -e > SRR_Acc_List.txt"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "1Wg-coeiyYEG"
      },
      "source": [
        "#add the file list after option file\n",
        "%%bash\n",
        "./prefetch --option-file SRR_Acc_List.txt \n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "k7Lqz69kPyb3"
      },
      "source": [
        "!mkdir /content/sample_data/studyx2"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "FMo6GE_py1_T"
      },
      "source": [
        "import os\n",
        "import shutil\n",
        "current_wkd=\"/content/sratoolkit.2.10.8-ubuntu64/bin\"\n",
        "dir_dst= \"/content/sample_data/studyx2/\"\n",
        "for root, dir, files in os.walk(top=current_wkd):\n",
        "    for file in files:\n",
        "        if file.endswith(\".sra\"):  #match files that match this extension\n",
        "            shutil.move(os.path.join(root, file), dir_dst)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "gyU1Pp6d9DPl"
      },
      "source": [
        ""
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "1rn5clg0g1_e"
      },
      "source": [
        "!mkdir /content/sample_data/study8/"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mrxkqL1t_pmu",
        "outputId": "357abc5f-6101-4474-cf5f-465ab63d342d"
      },
      "source": [
        "cd /content/sample_data/study1"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/sample_data/study1\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "jDynUq_2Yz-Q"
      },
      "source": [
        "rm *.gz"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "rKhc7oH7fc43"
      },
      "source": [
        "%%bash\n",
        "for f in in *.sra; do parallel-fastq-dump --sra-id --split-files  $f --threads 4  --outdir ~/ --gzip; done"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "yzKplnOibC69",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "05c239be-f3cd-44b4-8eda-cc5c69e09996"
      },
      "source": [
        "!cp /content/sample_data/study8/*.gz /content/drive/MyDrive/BeeTeam/study8/"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "cp: cannot stat '/content/sample_data/study8/*.gz': No such file or directory\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "6oopsmQ34Vmi"
      },
      "source": [
        ""
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Q5DSLP7C_w1v"
      },
      "source": [
        "#**1. Start here Download Qiime2**"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iuEwG2Qs1STx",
        "outputId": "b14bb795-dec0-428c-84ab-f11390e72154"
      },
      "source": [
        "!wget https://data.qiime2.org/distro/core/qiime2-2020.11-py36-linux-conda.yml"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "--2021-03-21 19:36:07--  https://data.qiime2.org/distro/core/qiime2-2020.11-py36-linux-conda.yml\n",
            "Resolving data.qiime2.org (data.qiime2.org)... 54.200.1.12\n",
            "Connecting to data.qiime2.org (data.qiime2.org)|54.200.1.12|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 FOUND\n",
            "Location: https://raw.githubusercontent.com/qiime2/environment-files/master/2020.11/release/qiime2-2020.11-py36-linux-conda.yml [following]\n",
            "--2021-03-21 19:36:08--  https://raw.githubusercontent.com/qiime2/environment-files/master/2020.11/release/qiime2-2020.11-py36-linux-conda.yml\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 9474 (9.3K) [text/plain]\n",
            "Saving to: ‘qiime2-2020.11-py36-linux-conda.yml’\n",
            "\n",
            "qiime2-2020.11-py36 100%[===================>]   9.25K  --.-KB/s    in 0s      \n",
            "\n",
            "2021-03-21 19:36:08 (47.8 MB/s) - ‘qiime2-2020.11-py36-linux-conda.yml’ saved [9474/9474]\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "fmQeQT0zADE6"
      },
      "source": [
        "#Create a conda enviroment for Qiime2"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "o22YcB-S2AGx",
        "outputId": "1c17e45f-8a64-42d7-8bc1-0cd575f45819"
      },
      "source": [
        "!conda env create -n qiime2-2020.11 --file qiime2-2020.11-py36-linux-conda.yml"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "ijson-3.1.3          | 24 KB     | : 100% 1.0/1 [00:00<00:00,  7.38it/s]               \n",
            "libglib-2.66.3       | 3.0 MB    | : 100% 1.0/1 [00:01<00:00,  1.26s/it]\n",
            "xorg-libsm-1.2.3     | 26 KB     | : 100% 1.0/1 [00:00<00:00, 16.32it/s]\n",
            "libgcc-7.2.0         | 304 KB    | : 100% 1.0/1 [00:00<00:00,  7.63it/s]\n",
            "libxml2-2.9.10       | 1.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.63it/s]\n",
            "fastcluster-1.1.26   | 45 KB     | : 100% 1.0/1 [00:00<00:00, 19.56it/s]\n",
            "ipywidgets-7.5.1     | 101 KB    | : 100% 1.0/1 [00:00<00:00, 13.30it/s]\n",
            "q2-quality-control-2 | 285 KB    | : 100% 1.0/1 [00:00<00:00,  4.52it/s]\n",
            "perl-parent-0.236    | 5 KB      | : 100% 1.0/1 [00:00<00:00, 26.19it/s]\n",
            "zlib-1.2.11          | 106 KB    | : 100% 1.0/1 [00:00<00:00, 15.05it/s]\n",
            "cutadapt-3.1         | 184 KB    | : 100% 1.0/1 [00:00<00:00, 10.03it/s]\n",
            "qiime2-2020.11.1     | 320 KB    | : 100% 1.0/1 [00:00<00:00,  2.74it/s]\n",
            "r-waldo-0.2.3        | 78 KB     | : 100% 1.0/1 [00:00<00:00, 14.60it/s]\n",
            "r-jsonlite-1.7.2     | 462 KB    | : 100% 1.0/1 [00:00<00:00,  7.83it/s]\n",
            "perl-scalar-list-uti | 38 KB     | : 100% 1.0/1 [00:00<00:00, 21.50it/s]\n",
            "python-3.6.12        | 38.4 MB   | : 100% 1.0/1 [00:09<00:00,  9.68s/it]\n",
            "r-stringr-1.4.0      | 209 KB    | : 100% 1.0/1 [00:00<00:00, 12.10it/s]\n",
            "isa-l-2.30.0         | 192 KB    | : 100% 1.0/1 [00:00<00:00, 12.20it/s]\n",
            "perl-business-isbn-3 | 15 KB     | : 100% 1.0/1 [00:00<00:00, 25.68it/s]\n",
            "ipython_genutils-0.2 | 21 KB     | : 100% 1.0/1 [00:00<00:00, 17.33it/s]\n",
            "xorg-xextproto-7.3.0 | 28 KB     | : 100% 1.0/1 [00:00<00:00, 20.08it/s]\n",
            "perl-digest-md5-2.55 | 18 KB     | : 100% 1.0/1 [00:00<00:00, 18.42it/s]\n",
            "more-itertools-8.6.0 | 37 KB     | : 100% 1.0/1 [00:00<00:00, 13.83it/s]\n",
            "r-mgcv-1.8_33        | 2.9 MB    | : 100% 1.0/1 [00:00<00:00,  1.58it/s]\n",
            "r-brio-1.1.0         | 37 KB     | : 100% 1.0/1 [00:00<00:00, 17.46it/s]\n",
            "r-lattice-0.20_41    | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  3.60it/s]\n",
            "kiwisolver-1.3.1     | 78 KB     | : 100% 1.0/1 [00:00<00:00, 17.93it/s]\n",
            "xorg-inputproto-2.3. | 19 KB     | : 100% 1.0/1 [00:00<00:00, 23.38it/s]\n",
            "xorg-libice-1.0.10   | 58 KB     | : 100% 1.0/1 [00:00<00:00, 19.77it/s]\n",
            "make-4.3             | 507 KB    | : 100% 1.0/1 [00:00<00:00,  3.48it/s]\n",
            "r-base-4.0.2         | 23.6 MB   | : 100% 1.0/1 [00:05<00:00,  5.47s/it]\n",
            "brotlipy-0.7.0       | 341 KB    | : 100% 1.0/1 [00:00<00:00,  8.79it/s]\n",
            "bioconductor-genomei | 4.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.20it/s]\n",
            "r-fansi-0.4.1        | 195 KB    | : 100% 1.0/1 [00:00<00:00, 13.84it/s]\n",
            "libev-4.33           | 104 KB    | : 100% 1.0/1 [00:00<00:00, 17.42it/s]\n",
            "perl-5.26.2          | 15.4 MB   | : 100% 1.0/1 [00:06<00:00,  6.56s/it]\n",
            "pandas-1.1.5         | 11.3 MB   | : 100% 1.0/1 [00:04<00:00,  4.76s/it]\n",
            "py-1.9.0             | 74 KB     | : 100% 1.0/1 [00:00<00:00, 12.98it/s]\n",
            "pyrsistent-0.17.3    | 89 KB     | : 100% 1.0/1 [00:00<00:00, 12.35it/s]\n",
            "tktable-2.10         | 89 KB     | : 100% 1.0/1 [00:00<00:00, 12.54it/s]\n",
            "perl-www-robotrules- | 12 KB     | : 100% 1.0/1 [00:00<00:00, 21.32it/s]\n",
            "emperor-1.0.1        | 621 KB    | : 100% 1.0/1 [00:00<00:00,  3.67it/s]\n",
            "q2-fragment-insertio | 31 KB     | : 100% 1.0/1 [00:00<00:00,  6.34it/s]\n",
            "r-bitops-1.0_6       | 39 KB     | : 100% 1.0/1 [00:00<00:00, 15.38it/s]\n",
            "alsa-lib-1.2.3       | 560 KB    | : 100% 1.0/1 [00:00<00:00,  5.10it/s]\n",
            "r-latticeextra-0.6_2 | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.27it/s]\n",
            "unifrac-0.20.1       | 356 KB    | : 100% 1.0/1 [00:00<00:00,  6.19it/s]\n",
            "msgpack-python-1.0.0 | 91 KB     | : 100% 1.0/1 [00:00<00:00, 13.95it/s]\n",
            "sina-1.7.1           | 1.9 MB    | : 100% 1.0/1 [00:00<00:00,  1.44it/s]\n",
            "perl-types-serialise | 10 KB     | : 100% 1.0/1 [00:00<00:00, 19.94it/s]\n",
            "biom-format-2.1.10   | 10.4 MB   | : 100% 1.0/1 [00:03<00:00,  3.49s/it]\n",
            "cffi-1.14.4          | 225 KB    | : 100% 1.0/1 [00:00<00:00,  8.76it/s]\n",
            "bioconductor-genomei | 7 KB      | : 100% 1.0/1 [00:00<00:00, 28.57it/s]\n",
            "future-0.18.2        | 716 KB    | : 100% 1.0/1 [00:00<00:00,  2.55it/s]\n",
            "_libgcc_mutex-0.1    | 3 KB      | : 100% 1.0/1 [00:00<00:00, 23.15it/s]\n",
            "typing_extensions-3. | 25 KB     | : 100% 1.0/1 [00:00<00:00, 21.93it/s]\n",
            "q2-dada2-2020.11.1   | 464 KB    | : 100% 1.0/1 [00:00<00:00,  3.16it/s]\n",
            "perl-xml-namespacesu | 11 KB     | : 100% 1.0/1 [00:00<00:00, 19.34it/s]\n",
            "urllib3-1.25.11      | 93 KB     | : 100% 1.0/1 [00:00<00:00,  8.69it/s]\n",
            "gxx_linux-64-7.5.0   | 23 KB     | : 100% 1.0/1 [00:00<00:00, 19.21it/s]\n",
            "r-bh-1.72.0_3        | 10.4 MB   | : 100% 1.0/1 [00:08<00:00,  8.33s/it]               \n",
            "lz4-3.1.1            | 44 KB     | : 100% 1.0/1 [00:00<00:00, 14.72it/s]\n",
            "ncurses-6.2          | 985 KB    | : 100% 1.0/1 [00:00<00:00,  1.37it/s]\n",
            "r-testthat-3.0.0     | 1.6 MB    | : 100% 1.0/1 [00:00<00:00,  2.64it/s]\n",
            "libtiff-4.1.0        | 618 KB    | : 100% 1.0/1 [00:00<00:00,  4.27it/s]\n",
            "threadpoolctl-2.1.0  | 15 KB     | : 100% 1.0/1 [00:00<00:00, 19.66it/s]\n",
            "perl-html-tagset-3.2 | 12 KB     | : 100% 1.0/1 [00:00<00:00, 15.68it/s]\n",
            "xorg-libx11-1.6.12   | 917 KB    | : 100% 1.0/1 [00:00<00:00,  3.25it/s]\n",
            "nbconvert-6.0.7      | 533 KB    | : 100% 1.0/1 [00:00<00:00,  3.89it/s]\n",
            "xorg-libxau-1.0.9    | 13 KB     | : 100% 1.0/1 [00:00<00:00, 18.58it/s]\n",
            "q2-diversity-lib-202 | 88 KB     | : 100% 1.0/1 [00:00<00:00,  4.92it/s]\n",
            "pandoc-2.11.2        | 17.7 MB   | : 100% 1.0/1 [00:09<00:00,  9.56s/it]\n",
            "xz-5.2.5             | 343 KB    | : 100% 1.0/1 [00:00<00:00,  6.71it/s]\n",
            "r-lambda.r-1.2.4     | 119 KB    | : 100% 1.0/1 [00:00<00:00, 17.22it/s]\n",
            "networkx-2.5         | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.65it/s]\n",
            "click-7.1.2          | 64 KB     | : 100% 1.0/1 [00:00<00:00, 17.02it/s]\n",
            "pickleshare-0.7.5    | 9 KB      | : 100% 1.0/1 [00:00<00:00, 23.18it/s]\n",
            "argon2-cffi-20.1.0   | 47 KB     | : 100% 1.0/1 [00:00<00:00, 16.50it/s]\n",
            "perl-io-html-1.001   | 13 KB     | : 100% 1.0/1 [00:00<00:00, 19.85it/s]\n",
            "r-pkgconfig-2.0.3    | 24 KB     | : 100% 1.0/1 [00:00<00:00, 21.38it/s]\n",
            "blast-2.10.1         | 20.6 MB   | : 100% 1.0/1 [00:06<00:00,  6.12s/it]               \n",
            "q2-longitudinal-2020 | 7.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.87s/it]\n",
            "q2-feature-classifie | 139 KB    | : 100% 1.0/1 [00:00<00:00,  4.06it/s]\n",
            "fribidi-1.0.10       | 112 KB    | : 100% 1.0/1 [00:00<00:00, 12.59it/s]\n",
            "pcre2-10.35          | 693 KB    | : 100% 1.0/1 [00:00<00:00,  3.36it/s]\n",
            "ca-certificates-2020 | 137 KB    | : 100% 1.0/1 [00:00<00:00, 12.47it/s]\n",
            "jinja2-2.11.2        | 93 KB     | : 100% 1.0/1 [00:00<00:00, 11.13it/s]\n",
            "tzlocal-2.1          | 18 KB     | : 100% 1.0/1 [00:00<00:00, 17.26it/s]\n",
            "perl-common-sense-3. | 11 KB     | : 100% 1.0/1 [00:00<00:00, 23.55it/s]\n",
            "perl-http-date-6.02  | 10 KB     | : 100% 1.0/1 [00:00<00:00, 22.87it/s]\n",
            "async_generator-1.10 | 18 KB     | : 100% 1.0/1 [00:00<00:00, 19.50it/s]\n",
            "requests-2.25.0      | 51 KB     | : 100% 1.0/1 [00:00<00:00, 18.14it/s]\n",
            "perl-uri-1.76        | 55 KB     | : 100% 1.0/1 [00:00<00:00, 15.25it/s]\n",
            "bioconductor-biocgen | 697 KB    | : 100% 1.0/1 [00:00<00:00,  4.16it/s]\n",
            "perl-lwp-protocol-ht | 9 KB      | : 100% 1.0/1 [00:00<00:00, 22.87it/s]\n",
            "r-matrix-1.2_18      | 3.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.24it/s]\n",
            "perl-http-cookies-6. | 19 KB     | : 100% 1.0/1 [00:00<00:00, 16.89it/s]\n",
            "perl-encode-2.88     | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.44it/s]\n",
            "readline-8.0         | 281 KB    | : 100% 1.0/1 [00:00<00:00,  6.67it/s]\n",
            "r-pkgload-1.1.0      | 167 KB    | : 100% 1.0/1 [00:00<00:00, 10.24it/s]\n",
            "zstd-1.4.5           | 712 KB    | : 100% 1.0/1 [00:00<00:00,  4.41it/s]\n",
            "xorg-libxfixes-5.0.3 | 17 KB     | : 100% 1.0/1 [00:00<00:00, 16.62it/s]\n",
            "lz4-c-1.9.2          | 203 KB    | : 100% 1.0/1 [00:00<00:00,  8.61it/s]\n",
            "pandocfilters-1.4.2  | 9 KB      | : 100% 1.0/1 [00:00<00:00, 21.56it/s]\n",
            "r-png-0.1_7          | 58 KB     | : 100% 1.0/1 [00:00<00:00, 19.47it/s]\n",
            "perl-io-socket-ssl-2 | 151 KB    | : 100% 1.0/1 [00:00<00:00, 11.44it/s]\n",
            "h5py-2.10.0          | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  2.20it/s]\n",
            "statsmodels-0.12.1   | 11.4 MB   | : 100% 1.0/1 [00:04<00:00,  4.16s/it]\n",
            "r-mass-7.3_53.1      | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  4.44it/s]\n",
            "r-diffobj-0.3.2      | 1.0 MB    | : 100% 1.0/1 [00:00<00:00,  4.16it/s]\n",
            "bioconductor-dada2-1 | 2.5 MB    | : 100% 1.0/1 [00:00<00:00,  1.85it/s]\n",
            "harfbuzz-2.7.2       | 1.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.71it/s]\n",
            "perl-file-listing-6. | 9 KB      | : 100% 1.0/1 [00:00<00:00, 25.10it/s]\n",
            "sed-4.8              | 264 KB    | : 100% 1.0/1 [00:00<00:00,  8.19it/s]\n",
            "notebook-6.1.5       | 6.3 MB    | : 100% 1.0/1 [00:02<00:00,  2.34s/it]\n",
            "r-nlme-3.1_150       | 2.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.99it/s]\n",
            "expat-2.2.9          | 163 KB    | : 100% 1.0/1 [00:00<00:00, 10.41it/s]\n",
            "bioconductor-zlibbio | 113 KB    | : 100% 1.0/1 [00:00<00:00, 11.21it/s]\n",
            "libcurl-7.71.1       | 312 KB    | : 100% 1.0/1 [00:00<00:00,  8.32it/s]\n",
            "r-rcolorbrewer-1.1_2 | 59 KB     | : 100% 1.0/1 [00:00<00:00, 21.22it/s]\n",
            "yaml-0.2.5           | 82 KB     | : 100% 1.0/1 [00:00<00:00, 10.51it/s]\n",
            "libssh2-1.9.0        | 225 KB    | : 100% 1.0/1 [00:00<00:00,  8.29it/s]\n",
            "bioconductor-biocpar | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  3.58it/s]\n",
            "pigz-2.3.4           | 76 KB     | : 100% 1.0/1 [00:00<00:00, 19.08it/s]\n",
            "r-farver-2.0.3       | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  3.05it/s]\n",
            "r-rprojroot-2.0.2    | 110 KB    | : 100% 1.0/1 [00:00<00:00, 16.04it/s]\n",
            "mafft-7.475          | 3.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.14s/it]              \n",
            "perl-exporter-tiny-1 | 23 KB     | : 100% 1.0/1 [00:00<00:00, 21.86it/s]\n",
            "r-desc-1.2.0         | 291 KB    | : 100% 1.0/1 [00:00<00:00, 13.79it/s]\n",
            "bioconductor-rhtslib | 1.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.61it/s]\n",
            "q2-emperor-2020.11.1 | 14 KB     | : 100% 1.0/1 [00:00<00:00,  6.64it/s]\n",
            "sortmerna-2.0        | 337 KB    | : 100% 1.0/1 [00:00<00:00,  8.91it/s]\n",
            "libgfortran-ng-7.5.0 | 22 KB     | : 100% 1.0/1 [00:00<00:00, 22.92it/s]\n",
            "decorator-4.4.2      | 11 KB     | : 100% 1.0/1 [00:00<00:00, 24.61it/s]\n",
            "libdeflate-1.6       | 60 KB     | : 100% 1.0/1 [00:00<00:00, 20.48it/s]\n",
            "q2-gneiss-2020.11.1  | 50 KB     | : 100% 1.0/1 [00:00<00:00,  6.04it/s]\n",
            "perl-archive-tar-2.3 | 31 KB     | : 100% 1.0/1 [00:00<00:00, 22.83it/s]\n",
            "r-tibble-3.0.4       | 384 KB    | : 100% 1.0/1 [00:00<00:00, 10.06it/s]\n",
            "libffi-3.3           | 51 KB     | : 100% 1.0/1 [00:00<00:00, 20.36it/s]\n",
            "bleach-3.2.1         | 111 KB    | : 100% 1.0/1 [00:00<00:00, 12.62it/s]\n",
            "perl-lwp-mediatypes- | 22 KB     | : 100% 1.0/1 [00:00<00:00, 18.79it/s]\n",
            "pyzmq-20.0.0         | 485 KB    | : 100% 1.0/1 [00:00<00:00,  3.68it/s]\n",
            "arb-bio-tools-6.0.6  | 673 KB    | : 100% 1.0/1 [00:00<00:00,  3.98it/s]\n",
            "matplotlib-base-3.2. | 7.2 MB    | : 100% 1.0/1 [00:02<00:00,  2.25s/it]\n",
            "r-scales-1.1.1       | 556 KB    | : 100% 1.0/1 [00:00<00:00,  7.04it/s]\n",
            "webencodings-0.5.1   | 12 KB     | : 100% 1.0/1 [00:00<00:00, 24.25it/s]\n",
            "entrypoints-0.3      | 8 KB      | : 100% 1.0/1 [00:00<00:00, 18.45it/s]\n",
            "r-rematch2-2.1.2     | 52 KB     | : 100% 1.0/1 [00:00<00:00, 19.80it/s]\n",
            "perl-list-moreutils- | 43 KB     | : 100% 1.0/1 [00:00<00:00, 19.02it/s]\n",
            "nbclient-0.5.1       | 60 KB     | : 100% 1.0/1 [00:00<00:00, 15.18it/s]\n",
            "perl-constant-1.33   | 10 KB     | : 100% 1.0/1 [00:00<00:00, 22.80it/s]\n",
            "pytest-6.1.2         | 418 KB    | : 100% 1.0/1 [00:00<00:00,  3.97it/s]\n",
            "q2-diversity-2020.11 | 225 KB    | : 100% 1.0/1 [00:00<00:00,  2.16it/s]\n",
            "perl-pathtools-3.75  | 41 KB     | : 100% 1.0/1 [00:00<00:00, 20.10it/s]\n",
            "r-prettyunits-1.1.1  | 40 KB     | : 100% 1.0/1 [00:00<00:00, 18.45it/s]\n",
            "iqtree-2.0.3         | 3.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.15it/s]\n",
            "r-futile.options-1.0 | 26 KB     | : 100% 1.0/1 [00:00<00:00, 20.23it/s]\n",
            "openjdk-11.0.8       | 173.3 MB  | : 100% 1.0/1 [00:44<00:00, 44.08s/it]               \n",
            "perl-business-isbn-d | 15 KB     | : 100% 1.0/1 [00:00<00:00,  9.70it/s]\n",
            "hdmedians-0.14.1     | 153 KB    | : 100% 1.0/1 [00:00<00:00, 14.66it/s]\n",
            "r-digest-0.6.27      | 213 KB    | : 100% 1.0/1 [00:00<00:00, 10.93it/s]\n",
            "widgetsnbextension-3 | 1.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.33it/s]\n",
            "perl-json-xs-2.34    | 63 KB     | : 100% 1.0/1 [00:00<00:00,  6.29it/s]                \n",
            "bzip2-1.0.8          | 484 KB    | : 100% 1.0/1 [00:00<00:00,  4.39it/s]\n",
            "perl-io-compress-2.0 | 81 KB     | : 100% 1.0/1 [00:00<00:00,  9.42it/s]\n",
            "wcwidth-0.2.5        | 33 KB     | : 100% 1.0/1 [00:00<00:00, 14.31it/s]\n",
            "cachecontrol-0.12.6  | 18 KB     | : 100% 1.0/1 [00:00<00:00, 21.79it/s]\n",
            "ipython-7.16.1       | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.66it/s]\n",
            "testpath-0.4.4       | 85 KB     | : 100% 1.0/1 [00:00<00:00, 18.81it/s]\n",
            "giflib-5.2.1         | 77 KB     | : 100% 1.0/1 [00:00<00:00, 18.84it/s]\n",
            "pysocks-1.7.1        | 27 KB     | : 100% 1.0/1 [00:00<00:00, 20.30it/s]\n",
            "pixman-0.40.0        | 627 KB    | : 100% 1.0/1 [00:00<00:00,  4.80it/s]\n",
            "xorg-libxrender-0.9. | 31 KB     | : 100% 1.0/1 [00:00<00:00, 19.61it/s]\n",
            "kernel-headers_linux | 707 KB    | : 100% 1.0/1 [00:00<00:00,  2.47it/s]\n",
            "gfortran_linux-64-7. | 22 KB     | : 100% 1.0/1 [00:00<00:00, 21.97it/s]\n",
            "perl-file-temp-0.230 | 26 KB     | : 100% 1.0/1 [00:00<00:00, 20.92it/s]\n",
            "r-munsell-0.5.0      | 246 KB    | : 100% 1.0/1 [00:00<00:00, 11.65it/s]\n",
            "importlib-metadata-3 | 12 KB     | : 100% 1.0/1 [00:00<00:00, 21.58it/s]\n",
            "r-pillar-1.4.7       | 194 KB    | : 100% 1.0/1 [00:00<00:00,  9.36it/s]\n",
            "markupsafe-1.1.1     | 27 KB     | : 100% 1.0/1 [00:00<00:00, 22.00it/s]\n",
            "nbformat-5.0.8       | 100 KB    | : 100% 1.0/1 [00:00<00:00, 12.50it/s]\n",
            "bioconductor-biostri | 13.9 MB   | : 100% 1.0/1 [00:02<00:00,  2.94s/it]\n",
            "tk-8.6.10            | 3.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.06s/it]\n",
            "libwebp-base-1.1.0   | 864 KB    | : 100% 1.0/1 [00:00<00:00,  3.10it/s]\n",
            "q2-cutadapt-2020.11. | 6.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.64s/it]\n",
            "entrez-direct-13.9   | 5.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.21s/it]\n",
            "r-vegan-2.5_6        | 3.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.38it/s]\n",
            "r-futile.logger-1.4. | 108 KB    | : 100% 1.0/1 [00:00<00:00, 16.60it/s]\n",
            "r-permute-0.9_5      | 507 KB    | : 100% 1.0/1 [00:00<00:00,  6.96it/s]\n",
            "libxcb-1.13          | 395 KB    | : 100% 1.0/1 [00:00<00:00,  4.50it/s]\n",
            "xorg-kbproto-1.0.7   | 27 KB     | : 100% 1.0/1 [00:00<00:00, 24.44it/s]\n",
            "q2-feature-table-202 | 227 KB    | : 100% 1.0/1 [00:00<00:00,  4.00it/s]\n",
            "fasttree-2.1.10      | 477 KB    | : 100% 1.0/1 [00:00<00:00,  4.70it/s]\n",
            "r-jpeg-0.1_8.1       | 51 KB     | : 100% 1.0/1 [00:00<00:00, 19.72it/s]\n",
            "certifi-2020.12.5    | 143 KB    | : 100% 1.0/1 [00:00<00:00, 15.63it/s]\n",
            "r-formatr-1.7        | 166 KB    | : 100% 1.0/1 [00:00<00:00, 13.59it/s]\n",
            "r-rcurl-1.98_1.2     | 960 KB    | : 100% 1.0/1 [00:00<00:00,  3.67it/s]\n",
            "ld_impl_linux-64-2.3 | 618 KB    | : 100% 1.0/1 [00:00<00:00,  3.32it/s]\n",
            "pexpect-4.8.0        | 47 KB     | : 100% 1.0/1 [00:00<00:00, 20.01it/s]\n",
            "sysroot_linux-64-2.1 | 30.2 MB   | : 100% 1.0/1 [00:08<00:00,  8.21s/it]\n",
            "perl-http-negotiate- | 13 KB     | : 100% 1.0/1 [00:00<00:00, 19.53it/s]\n",
            "blas-2.17            | 11 KB     | : 100% 1.0/1 [00:00<00:00, 26.27it/s]\n",
            "bioconductor-s4vecto | 2.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.99it/s]\n",
            "r-glue-1.4.2         | 141 KB    | : 100% 1.0/1 [00:00<00:00,  5.93it/s]                \n",
            "_r-mutex-1.0.1       | 3 KB      | : 100% 1.0/1 [00:00<00:00, 19.44it/s]\n",
            "c-ares-1.17.1        | 109 KB    | : 100% 1.0/1 [00:00<00:00, 13.44it/s]\n",
            "perl-io-zlib-1.10    | 9 KB      | : 100% 1.0/1 [00:00<00:00, 21.20it/s]\n",
            "jupyter_core-4.7.0   | 72 KB     | : 100% 1.0/1 [00:00<00:00, 11.06it/s]\n",
            "gfortran_impl_linux- | 9.0 MB    | : 100% 1.0/1 [00:02<00:00,  2.69s/it]\n",
            "scikit-learn-0.23.1  | 6.8 MB    | : 100% 1.0/1 [00:02<00:00,  2.92s/it]\n",
            "perl-html-parser-3.7 | 66 KB     | : 100% 1.0/1 [00:00<00:00, 14.32it/s]\n",
            "joblib-0.17.0        | 204 KB    | : 100% 1.0/1 [00:00<00:00,  7.13it/s]\n",
            "pluggy-0.13.1        | 29 KB     | : 100% 1.0/1 [00:00<00:00, 18.65it/s]\n",
            "libgomp-9.3.0        | 376 KB    | : 100% 1.0/1 [00:00<00:00,  7.08it/s]\n",
            "xorg-recordproto-1.1 | 7 KB      | : 100% 1.0/1 [00:00<00:00, 22.34it/s]\n",
            "gcc_impl_linux-64-7. | 38.3 MB   | : 100% 1.0/1 [00:11<00:00, 11.37s/it]               \n",
            "dnaio-0.4.4          | 136 KB    | : 100% 1.0/1 [00:00<00:00, 11.65it/s]\n",
            "r-rcppparallel-5.0.2 | 1.6 MB    | : 100% 1.0/1 [00:00<00:00,  2.06it/s]\n",
            "r-stringi-1.5.3      | 805 KB    | : 100% 1.0/1 [00:00<00:00,  5.67it/s]\n",
            "r-lifecycle-0.2.0    | 112 KB    | : 100% 1.0/1 [00:00<00:00, 13.18it/s]\n",
            "jsonschema-3.2.0     | 45 KB     | : 100% 1.0/1 [00:00<00:00, 11.36it/s]\n",
            "hdf5-1.10.5          | 3.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.15it/s]\n",
            "wheel-0.36.1         | 30 KB     | : 100% 1.0/1 [00:00<00:00, 21.85it/s]\n",
            "tbb-2020.2           | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  2.73it/s]\n",
            "samtools-1.11        | 383 KB    | : 100% 1.0/1 [00:00<00:00,  7.20it/s]\n",
            "sepp-4.3.10          | 6.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.64s/it]\n",
            "glib-2.66.3          | 502 KB    | : 100% 1.0/1 [00:00<00:00,  3.29it/s]\n",
            "r-hwriter-1.3.2      | 173 KB    | : 100% 1.0/1 [00:00<00:00, 14.50it/s]\n",
            "r-backports-1.2.1    | 99 KB     | : 100% 1.0/1 [00:00<00:00, 17.20it/s]\n",
            "jpeg-9d              | 264 KB    | : 100% 1.0/1 [00:00<00:00,  7.29it/s]\n",
            "perl-net-http-6.19   | 19 KB     | : 100% 1.0/1 [00:00<00:00, 18.94it/s]\n",
            "ptyprocess-0.6.0     | 15 KB     | : 100% 1.0/1 [00:00<00:00, 24.78it/s]\n",
            "r-cluster-2.1.0      | 553 KB    | : 100% 1.0/1 [00:00<00:00,  6.38it/s]\n",
            "perl-try-tiny-0.30   | 17 KB     | : 100% 1.0/1 [00:00<00:00, 17.87it/s]\n",
            "prometheus_client-0. | 44 KB     | : 100% 1.0/1 [00:00<00:00, 15.97it/s]\n",
            "krb5-1.17.2          | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.94it/s]\n",
            "libblas-3.8.0        | 11 KB     | : 100% 1.0/1 [00:00<00:00, 25.45it/s]\n",
            "perl-encode-locale-1 | 11 KB     | : 100% 1.0/1 [00:00<00:00, 24.96it/s]\n",
            "perl-time-local-1.28 | 10 KB     | : 100% 1.0/1 [00:00<00:00, 23.61it/s]\n",
            "_openmp_mutex-4.5    | 5 KB      | : 100% 1.0/1 [00:00<00:00, 23.69it/s]\n",
            "sqlite-3.34.0        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.89it/s]\n",
            "q2-deblur-2020.11.1  | 20.4 MB   | : 100% 1.0/1 [00:03<00:00,  3.95s/it]\n",
            "libsodium-1.0.18     | 366 KB    | : 100% 1.0/1 [00:00<00:00,  7.55it/s]\n",
            "backports.functools_ | 8 KB      | : 100% 1.0/1 [00:00<00:00, 24.38it/s]\n",
            "q2-demux-2020.11.1   | 191 KB    | : 100% 1.0/1 [00:00<00:00,  4.52it/s]\n",
            "binutils_linux-64-2. | 22 KB     | : 100% 1.0/1 [00:00<00:00, 23.32it/s]\n",
            "bowtie2-2.4.2        | 12.7 MB   | : 100% 1.0/1 [00:03<00:00,  3.10s/it]\n",
            "graphite2-1.3.13     | 102 KB    | : 100% 1.0/1 [00:00<00:00, 16.85it/s]\n",
            "parso-0.7.1          | 70 KB     | : 100% 1.0/1 [00:00<00:00, 16.59it/s]\n",
            "cython-0.29.21       | 2.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.15it/s]\n",
            "jupyterlab_pygments- | 8 KB      | : 100% 1.0/1 [00:00<00:00, 24.23it/s]\n",
            "jupyter_client-6.1.7 | 76 KB     | : 100% 1.0/1 [00:00<00:00, 15.97it/s]\n",
            "perl-list-moreutils- | 28 KB     | : 100% 1.0/1 [00:00<00:00, 19.02it/s]\n",
            "perl-file-path-2.16  | 17 KB     | : 100% 1.0/1 [00:00<00:00, 19.20it/s]\n",
            "q2-metadata-2020.11. | 967 KB    | : 100% 1.0/1 [00:00<00:00,  2.48it/s]\n",
            "xorg-libxi-1.7.10    | 45 KB     | : 100% 1.0/1 [00:00<00:00, 16.31it/s]\n",
            "jedi-0.17.2          | 957 KB    | : 100% 1.0/1 [00:00<00:00,  1.23it/s]\n",
            "backcall-0.2.0       | 13 KB     | : 100% 1.0/1 [00:00<00:00, 23.61it/s]\n",
            "r-ps-1.5.0           | 290 KB    | : 100% 1.0/1 [00:00<00:00, 11.18it/s]\n",
            "r-pkgbuild-1.1.0     | 156 KB    | : 100% 1.0/1 [00:00<00:00, 16.52it/s]\n",
            "bioconductor-rsamtoo | 3.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.13it/s]\n",
            "r-ggplot2-3.3.2      | 3.9 MB    | : 100% 1.0/1 [00:00<00:00,  1.10it/s]\n",
            "perl-socket-2.027    | 31 KB     | : 100% 1.0/1 [00:00<00:00, 20.50it/s]\n",
            "r-vctrs-0.3.5        | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  3.57it/s]\n",
            "r-r6-2.5.0           | 89 KB     | : 100% 1.0/1 [00:00<00:00, 17.93it/s]\n",
            "pcre-8.44            | 261 KB    | : 100% 1.0/1 [00:00<00:00,  8.30it/s]\n",
            "libstdcxx-devel_linu | 9.3 MB    | : 100% 1.0/1 [00:02<00:00,  2.80s/it]\n",
            "numpy-1.19.4         | 5.3 MB    | : 100% 1.0/1 [00:02<00:00,  2.06s/it]\n",
            "bibtexparser-1.1.0   | 32 KB     | : 100% 1.0/1 [00:00<00:00, 19.32it/s]\n",
            "r-praise-1.0.0       | 23 KB     | : 100% 1.0/1 [00:00<00:00, 21.29it/s]\n",
            "perl-libwww-perl-6.3 | 94 KB     | : 100% 1.0/1 [00:00<00:00,  7.99it/s]\n",
            "toml-0.10.2          | 18 KB     | : 100% 1.0/1 [00:00<00:00, 20.87it/s]\n",
            "perl-exporter-5.72   | 13 KB     | : 100% 1.0/1 [00:00<00:00, 24.97it/s]\n",
            "perl-compress-raw-zl | 72 KB     | : 100% 1.0/1 [00:00<00:00, 18.39it/s]\n",
            "nose-1.3.7           | 118 KB    | : 100% 1.0/1 [00:00<00:00, 10.63it/s]\n",
            "r-matrixstats-0.57.0 | 918 KB    | : 100% 1.0/1 [00:00<00:00,  3.24it/s]\n",
            "bioconductor-genomic | 2.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.71it/s]\n",
            "terminado-0.9.1      | 25 KB     | : 100% 1.0/1 [00:00<00:00, 22.97it/s]\n",
            "perl-html-tree-5.07  | 129 KB    | : 100% 1.0/1 [00:00<00:00, 10.38it/s]\n",
            "r-labeling-0.4.2     | 67 KB     | : 100% 1.0/1 [00:00<00:00, 19.64it/s]\n",
            "cryptography-3.3.1   | 624 KB    | : 100% 1.0/1 [00:00<00:00,  2.67it/s]\n",
            "perl-xml-sax-1.02    | 36 KB     | : 100% 1.0/1 [00:00<00:00,  9.51it/s]\n",
            "perl-mozilla-ca-2018 | 140 KB    | : 100% 1.0/1 [00:00<00:00, 15.59it/s]\n",
            "gsl-2.6              | 3.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.49s/it]\n",
            "freetype-2.10.4      | 890 KB    | : 100% 1.0/1 [00:00<00:00,  3.22it/s]\n",
            "perl-compress-raw-bz | 44 KB     | : 100% 1.0/1 [00:00<00:00, 17.53it/s]\n",
            "bioconductor-iranges | 2.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.75it/s]\n",
            "liblapack-3.8.0      | 11 KB     | : 100% 1.0/1 [00:00<00:00, 24.04it/s]\n",
            "perl-extutils-makema | 153 KB    | : 100% 1.0/1 [00:00<00:00,  9.72it/s]\n",
            "gneiss-0.4.6         | 41 KB     | : 100% 1.0/1 [00:00<00:00, 17.37it/s]\n",
            "r-utf8-1.1.4         | 161 KB    | : 100% 1.0/1 [00:00<00:00, 14.20it/s]\n",
            "gettext-0.19.8.1     | 3.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.53s/it]\n",
            "r-rstudioapi-0.13    | 281 KB    | : 100% 1.0/1 [00:00<00:00, 10.93it/s]\n",
            "pyparsing-2.4.7      | 60 KB     | : 100% 1.0/1 [00:00<00:00, 19.96it/s]\n",
            "pygments-2.7.3       | 708 KB    | : 100% 1.0/1 [00:00<00:00,  3.15it/s]\n",
            "perl-http-message-6. | 51 KB     | : 100% 1.0/1 [00:00<00:00, 17.33it/s]\n",
            "deblur-1.1.0         | 2.3 MB    | : 100% 1.0/1 [00:02<00:00,  2.23s/it]\n",
            "fontconfig-2.13.1    | 344 KB    | : 100% 1.0/1 [00:00<00:00,  6.21it/s]\n",
            "curl-7.71.1          | 139 KB    | : 100% 1.0/1 [00:00<00:00, 14.23it/s]\n",
            "libopenblas-0.3.10   | 7.8 MB    | : 100% 1.0/1 [00:02<00:00,  2.44s/it]\n",
            "xorg-libxdmcp-1.1.3  | 19 KB     | : 100% 1.0/1 [00:00<00:00, 10.74it/s]\n",
            "scikit-bio-0.5.6     | 1.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.33it/s]\n",
            "libcblas-3.8.0       | 11 KB     | : 100% 1.0/1 [00:00<00:00, 20.03it/s]\n",
            "xopen-1.0.1          | 19 KB     | : 100% 1.0/1 [00:00<00:00, 21.89it/s]\n",
            "r-viridislite-0.3.0  | 64 KB     | : 100% 1.0/1 [00:00<00:00, 19.75it/s]\n",
            "packaging-20.7       | 32 KB     | : 100% 1.0/1 [00:00<00:00, 21.47it/s]\n",
            "gcc_linux-64-7.5.0   | 23 KB     | : 100% 1.0/1 [00:00<00:00, 15.36it/s]\n",
            "xorg-libxtst-1.2.3   | 31 KB     | : 100% 1.0/1 [00:00<00:00, 19.62it/s]\n",
            "xorg-renderproto-0.1 | 9 KB      | : 100% 1.0/1 [00:00<00:00, 25.08it/s]\n",
            "q2-sample-classifier | 626 KB    | : 100% 1.0/1 [00:00<00:00,  3.32it/s]\n",
            "six-1.15.0           | 14 KB     | : 100% 1.0/1 [00:00<00:00, 24.40it/s]\n",
            "pyyaml-5.3.1         | 198 KB    | : 100% 1.0/1 [00:00<00:00,  9.31it/s]\n",
            "r-snow-0.4_3         | 123 KB    | : 100% 1.0/1 [00:00<00:00, 15.37it/s]\n",
            "q2cli-2020.11.1      | 77 KB     | : 100% 1.0/1 [00:00<00:00,  3.76it/s]\n",
            "r-reshape2-1.4.4     | 135 KB    | : 100% 1.0/1 [00:00<00:00, 14.99it/s]\n",
            "perl-carp-1.38       | 15 KB     | : 100% 1.0/1 [00:00<00:00, 14.81it/s]\n",
            "libpng-1.6.37        | 306 KB    | : 100% 1.0/1 [00:00<00:00,  6.90it/s]\n",
            "q2-vsearch-2020.11.1 | 634 KB    | : 100% 1.0/1 [00:00<00:00,  3.11it/s]\n",
            "boost-cpp-1.70.0     | 21.1 MB   | : 100% 1.0/1 [00:13<00:00, 13.57s/it]               \n",
            "libstdcxx-ng-9.3.0   | 4.0 MB    | : 100% 1.0/1 [00:01<00:00,  1.15s/it]\n",
            "backports-1.0        | 4 KB      | : 100% 1.0/1 [00:00<00:00, 21.61it/s]\n",
            "nest-asyncio-1.4.3   | 9 KB      | : 100% 1.0/1 [00:00<00:00, 20.95it/s]\n",
            "setuptools-49.6.0    | 936 KB    | : 100% 1.0/1 [00:00<00:00,  2.10it/s]\n",
            "bioconductor-summari | 2.7 MB    | : 100% 1.0/1 [00:00<00:00,  1.53it/s]\n",
            "bioconductor-biobase | 2.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.83it/s]\n",
            "natsort-7.1.0        | 33 KB     | : 100% 1.0/1 [00:00<00:00, 18.50it/s]\n",
            "r-isoband-0.2.3      | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.06it/s]\n",
            "perl-net-ssleay-1.88 | 289 KB    | : 100% 1.0/1 [00:00<00:00,  6.49it/s]\n",
            "r-gtable-0.3.0       | 423 KB    | : 100% 1.0/1 [00:00<00:00,  6.06it/s]\n",
            "perl-app-cpanminus-1 | 234 KB    | : 100% 1.0/1 [00:00<00:00,  7.63it/s]\n",
            "perl-xml-sax-expat-0 | 10 KB     | : 100% 1.0/1 [00:00<00:00, 23.11it/s]\n",
            "r-colorspace-2.0_0   | 2.5 MB    | : 100% 1.0/1 [00:00<00:00,  1.53it/s]\n",
            "r-withr-2.3.0        | 216 KB    | : 100% 1.0/1 [00:00<00:00, 11.67it/s]\n",
            "raxml-8.2.12         | 3.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.29it/s]\n",
            "bioconductor-genomic | 2.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.79it/s]\n",
            "r-rlang-0.4.9        | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  4.22it/s]\n",
            "perl-http-daemon-6.0 | 14 KB     | : 100% 1.0/1 [00:00<00:00, 18.29it/s]\n",
            "bioconductor-matrixg | 321 KB    | : 100% 1.0/1 [00:00<00:00,  9.11it/s]\n",
            "icu-67.1             | 12.9 MB   | : 100% 1.0/1 [00:03<00:00,  3.71s/it]\n",
            "r-cli-2.2.0          | 443 KB    | : 100% 1.0/1 [00:00<00:00,  7.58it/s]\n",
            "libgcc-ng-9.3.0      | 7.8 MB    | : 100% 1.0/1 [00:02<00:00,  2.01s/it]               \n",
            "iniconfig-1.1.1      | 8 KB      | : 100% 1.0/1 [00:00<00:00, 17.54it/s]\n",
            "pyopenssl-20.0.0     | 48 KB     | : 100% 1.0/1 [00:00<00:00, 15.43it/s]\n",
            "r-plyr-1.8.6         | 830 KB    | : 100% 1.0/1 [00:00<00:00,  4.37it/s]\n",
            "python-dateutil-2.8. | 220 KB    | : 100% 1.0/1 [00:00<00:00, 13.48it/s]\n",
            "perl-mime-base64-3.1 | 15 KB     | : 100% 1.0/1 [00:00<00:00, 21.63it/s]\n",
            "perl-data-dumper-2.1 | 35 KB     | : 100% 1.0/1 [00:00<00:00, 18.95it/s]\n",
            "liblapacke-3.8.0     | 11 KB     | : 100% 1.0/1 [00:00<00:00, 17.68it/s]\n",
            "defusedxml-0.6.0     | 22 KB     | : 100% 1.0/1 [00:00<00:00, 19.88it/s]\n",
            "q2-types-2020.11.1   | 111 KB    | : 100% 1.0/1 [00:00<00:00,  3.82it/s]\n",
            "r-crayon-1.3.4       | 747 KB    | : 100% 1.0/1 [00:00<00:00,  4.43it/s]\n",
            "zeromq-4.3.3         | 288 KB    | : 100% 1.0/1 [00:00<00:00, 10.10it/s]\n",
            "tornado-6.1          | 643 KB    | : 100% 1.0/1 [00:00<00:00,  2.85it/s]\n",
            "pycparser-2.20       | 94 KB     | : 100% 1.0/1 [00:00<00:00, 12.19it/s]\n",
            "libgfortran4-7.5.0   | 1.3 MB    | : 100% 1.0/1 [00:00<00:00,  2.17it/s]\n",
            "r-rcpp-1.0.5         | 2.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.31it/s]\n",
            "bokeh-2.2.3          | 7.0 MB    | : 100% 1.0/1 [00:03<00:00,  3.47s/it]\n",
            "hmmer-3.1b2          | 5.5 MB    | : 100% 1.0/1 [00:01<00:00,  1.61s/it]\n",
            "perl-json-4.02       | 52 KB     | : 100% 1.0/1 [00:00<00:00, 18.66it/s]\n",
            "idna-2.10            | 52 KB     | : 100% 1.0/1 [00:00<00:00, 18.04it/s]\n",
            "xorg-xproto-7.0.31   | 73 KB     | : 100% 1.0/1 [00:00<00:00, 16.40it/s]\n",
            "importlib_metadata-3 | 3 KB      | : 100% 1.0/1 [00:00<00:00, 22.25it/s]\n",
            "gxx_impl_linux-64-7. | 9.5 MB    | : 100% 1.0/1 [00:02<00:00,  2.97s/it]               \n",
            "mistune-0.8.4        | 54 KB     | : 100% 1.0/1 [00:00<00:00, 14.52it/s]\n",
            "traitlets-4.3.3      | 133 KB    | : 100% 1.0/1 [00:00<00:00, 10.39it/s]\n",
            "bwidget-1.9.14       | 119 KB    | : 100% 1.0/1 [00:00<00:00,  9.56it/s]\n",
            "ipykernel-5.3.4      | 166 KB    | : 100% 1.0/1 [00:00<00:00,  8.19it/s]\n",
            "q2-taxa-2020.11.1    | 108 KB    | : 100% 1.0/1 [00:00<00:00,  4.82it/s]\n",
            "chardet-3.0.4        | 194 KB    | : 100% 1.0/1 [00:00<00:00,  9.13it/s]\n",
            "r-zeallot-0.1.0      | 62 KB     | : 100% 1.0/1 [00:00<00:00, 17.92it/s]\n",
            "r-processx-3.4.5     | 296 KB    | : 100% 1.0/1 [00:00<00:00, 12.04it/s]\n",
            "perl-storable-3.15   | 63 KB     | : 100% 1.0/1 [00:00<00:00, 19.68it/s]\n",
            "seaborn-base-0.11.0  | 216 KB    | : 100% 1.0/1 [00:00<00:00,  4.54it/s]\n",
            "libgcc-devel_linux-6 | 3.6 MB    | : 100% 1.0/1 [00:00<00:00,  1.08it/s]\n",
            "pthread-stubs-0.4    | 5 KB      | : 100% 1.0/1 [00:00<00:00, 24.29it/s]\n",
            "q2-alignment-2020.11 | 19 KB     | : 100% 1.0/1 [00:00<00:00,  4.94it/s]\n",
            "q2-phylogeny-2020.11 | 46 KB     | : 100% 1.0/1 [00:00<00:00,  5.77it/s]\n",
            "q2templates-2020.11. | 188 KB    | : 100% 1.0/1 [00:00<00:00,  4.63it/s]\n",
            "send2trash-1.5.0     | 12 KB     | : 100% 1.0/1 [00:00<00:00, 23.03it/s]\n",
            "r-ellipsis-0.3.1     | 49 KB     | : 100% 1.0/1 [00:00<00:00, 16.36it/s]\n",
            "bioconductor-shortre | 5.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.18s/it]\n",
            "pytz-2020.4          | 229 KB    | : 100% 1.0/1 [00:00<00:00,  5.19it/s]\n",
            "lockfile-0.12.2      | 11 KB     | : 100% 1.0/1 [00:00<00:00, 19.10it/s]\n",
            "cycler-0.10.0        | 9 KB      | : 100% 1.0/1 [00:00<00:00, 26.22it/s]\n",
            "pillow-8.0.1         | 687 KB    | : 100% 1.0/1 [00:00<00:00,  3.82it/s]\n",
            "q2-quality-filter-20 | 32 KB     | : 100% 1.0/1 [00:00<00:00,  6.40it/s]\n",
            "vsearch-2.7.0        | 713 KB    | : 100% 1.0/1 [00:00<00:00,  4.50it/s]\n",
            "llvm-openmp-11.0.0   | 2.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.34it/s]\n",
            "libarbdb-6.0.6       | 290 KB    | : 100% 1.0/1 [00:00<00:00,  8.96it/s]\n",
            "libedit-3.1.20191231 | 121 KB    | : 100% 1.0/1 [00:00<00:00, 13.56it/s]\n",
            "htslib-1.11          | 1.7 MB    | : 100% 1.0/1 [00:00<00:00,  1.65it/s]\n",
            "scipy-1.5.3          | 18.6 MB   | : 100% 1.0/1 [00:05<00:00,  5.95s/it]             \n",
            "cairo-1.16.0         | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  1.51it/s]\n",
            "libiconv-1.16        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.92it/s]\n",
            "python_abi-3.6       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 24.59it/s]\n",
            "perl-xml-simple-2.25 | 48 KB     | : 100% 1.0/1 [00:00<00:00, 17.96it/s]\n",
            "seaborn-0.11.0       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 23.00it/s]\n",
            "attrs-20.3.0         | 41 KB     | : 100% 1.0/1 [00:00<00:00, 17.40it/s]\n",
            "olefile-0.46         | 32 KB     | : 100% 1.0/1 [00:00<00:00, 21.42it/s]\n",
            "matplotlib-3.2.2     | 6 KB      | : 100% 1.0/1 [00:00<00:00, 27.99it/s]\n",
            "psutil-5.7.3         | 338 KB    | : 100% 1.0/1 [00:00<00:00,  4.61it/s]\n",
            "bioconductor-xvector | 730 KB    | : 100% 1.0/1 [00:00<00:00,  4.44it/s]\n",
            "lcms2-2.11           | 434 KB    | : 100% 1.0/1 [00:00<00:00,  5.57it/s]\n",
            "pip-20.3.1           | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.83it/s]\n",
            "libuuid-2.32.1       | 28 KB     | : 100% 1.0/1 [00:00<00:00,  7.72it/s]              \n",
            "perl-xsloader-0.24   | 8 KB      | : 100% 1.0/1 [00:00<00:00, 15.48it/s]\n",
            "prompt-toolkit-3.0.8 | 237 KB    | : 100% 1.0/1 [00:00<00:00,  4.33it/s]\n",
            "dendropy-4.5.1       | 304 KB    | : 100% 1.0/1 [00:00<00:00,  4.34it/s]\n",
            "patsy-0.5.1          | 187 KB    | : 100% 1.0/1 [00:00<00:00, 11.13it/s]\n",
            "perl-test-requiresin | 7 KB      | : 100% 1.0/1 [00:00<00:00, 23.70it/s]\n",
            "openssl-1.1.1h       | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.48it/s]\n",
            "zipp-3.4.0           | 11 KB     | : 100% 1.0/1 [00:00<00:00, 19.89it/s]\n",
            "perl-base-2.23       | 10 KB     | : 100% 1.0/1 [00:00<00:00, 21.01it/s]\n",
            "q2-composition-2020. | 169 KB    | : 100% 1.0/1 [00:00<00:00,  4.63it/s]\n",
            "libnghttp2-1.41.0    | 774 KB    | : 100% 1.0/1 [00:00<00:00,  4.02it/s]\n",
            "xorg-fixesproto-5.0  | 8 KB      | : 100% 1.0/1 [00:00<00:00, 15.53it/s]\n",
            "bioconductor-delayed | 2.3 MB    | : 100% 1.0/1 [00:00<00:00,  1.77it/s]\n",
            "pango-1.42.4         | 533 KB    | : 100% 1.0/1 [00:00<00:00,  3.30it/s]\n",
            "xorg-libxext-1.3.4   | 51 KB     | : 100% 1.0/1 [00:00<00:00, 20.52it/s]\n",
            "r-magrittr-2.0.1     | 211 KB    | : 100% 1.0/1 [00:00<00:00, 12.74it/s]\n",
            "binutils_impl_linux- | 9.3 MB    | : 100% 1.0/1 [00:02<00:00,  2.69s/it]               \n",
            "perl-xml-parser-2.44 | 165 KB    | : 100% 1.0/1 [00:00<00:00,  8.66it/s]\n",
            "perl-digest-hmac-1.0 | 9 KB      | : 100% 1.0/1 [00:00<00:00, 19.35it/s]\n",
            "perl-xml-sax-base-1. | 19 KB     | : 100% 1.0/1 [00:00<00:00, 15.32it/s]\n",
            "r-assertthat-0.2.1   | 70 KB     | : 100% 1.0/1 [00:00<00:00,  5.08it/s]                \n",
            "r-callr-3.5.1        | 388 KB    | : 100% 1.0/1 [00:00<00:00,  9.23it/s]\n",
            "r-evaluate-0.14      | 81 KB     | : 100% 1.0/1 [00:00<00:00, 14.61it/s]\n",
            "perl-ntlm-1.09       | 15 KB     | : 100% 1.0/1 [00:00<00:00, 21.81it/s]\n",
            "Preparing transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Verifying transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ Enabling notebook extension jupyter-js-widgets/extension...\n",
            "Paths used for configuration of notebook: \n",
            "    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.d/widgetsnbextension.json\n",
            "    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.json\n",
            "Paths used for configuration of notebook: \n",
            "    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.d/widgetsnbextension.json\n",
            "      - Validating: \u001b[32mOK\u001b[0m\n",
            "Paths used for configuration of notebook: \n",
            "    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.d/widgetsnbextension.json\n",
            "    \t/usr/local/envs/qiime2-2020.11/etc/jupyter/nbconfig/notebook.json\n",
            "\n",
            "\b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "#\n",
            "# To activate this environment, use\n",
            "#\n",
            "#     $ conda activate qiime2-2020.11\n",
            "#\n",
            "# To deactivate an active environment, use\n",
            "#\n",
            "#     $ conda deactivate\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "39M8EoI-Ahie"
      },
      "source": [
        "#Test Qiime2, all qiime2 commands need to be executed with \"!source activate qiime2-2020.11 &&\""
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nQW9Fd0E4v23",
        "outputId": "0220a6e0-b052-46e1-cbdf-62633c6fe6ef"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime --help"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Usage: \u001b[34mqiime\u001b[0m [OPTIONS] COMMAND [ARGS]...\n",
            "\n",
            "  QIIME 2 command-line interface (q2cli)\n",
            "  --------------------------------------\n",
            "\n",
            "  To get help with QIIME 2, visit\n",
            "  https://qiime2.org.\n",
            "\n",
            "  To enable tab completion in Bash, run the\n",
            "  following command or add it to your\n",
            "  .bashrc/.bash_profile:\n",
            "\n",
            "      source tab-qiime\n",
            "\n",
            "  To enable tab completion in ZSH, run the\n",
            "  following commands or add them to your .zshrc:\n",
            "\n",
            "      autoload bashcompinit && bashcompinit &&\n",
            "      source tab-qiime\n",
            "\n",
            "\u001b[1mOptions\u001b[0m:\n",
            "  \u001b[34m--version\u001b[0m   Show the version and exit.\n",
            "  \u001b[34m--help\u001b[0m      Show this message and exit.\n",
            "\n",
            "\u001b[1mCommands\u001b[0m:\n",
            "  \u001b[34minfo\u001b[0m                Display information about...\n",
            "  \u001b[34mtools\u001b[0m               Tools for working with...\n",
            "  \u001b[34mdev\u001b[0m                 Utilities for developers...\n",
            "  \u001b[34malignment\u001b[0m           Plugin for generating and\n",
            "                      manipulating alignments.\n",
            "\n",
            "  \u001b[34mcomposition\u001b[0m         Plugin for compositional\n",
            "                      data analysis.\n",
            "\n",
            "  \u001b[34mcutadapt\u001b[0m            Plugin for removing adapter\n",
            "                      sequences, primers, and\n",
            "                      other unwanted sequence from\n",
            "                      sequence data.\n",
            "\n",
            "  \u001b[34mdada2\u001b[0m               Plugin for sequence quality\n",
            "                      control with DADA2.\n",
            "\n",
            "  \u001b[34mdeblur\u001b[0m              Plugin for sequence quality\n",
            "                      control with Deblur.\n",
            "\n",
            "  \u001b[34mdemux\u001b[0m               Plugin for demultiplexing &\n",
            "                      viewing sequence quality.\n",
            "\n",
            "  \u001b[34mdiversity\u001b[0m           Plugin for exploring\n",
            "                      community diversity.\n",
            "\n",
            "  \u001b[34mdiversity-lib\u001b[0m       Plugin for computing\n",
            "                      community diversity.\n",
            "\n",
            "  \u001b[34memperor\u001b[0m             Plugin for ordination\n",
            "                      plotting with Emperor.\n",
            "\n",
            "  \u001b[34mfeature-classifier\u001b[0m  Plugin for taxonomic\n",
            "                      classification.\n",
            "\n",
            "  \u001b[34mfeature-table\u001b[0m       Plugin for working with\n",
            "                      sample by feature tables.\n",
            "\n",
            "  \u001b[34mfragment-insertion\u001b[0m  Plugin for extending\n",
            "                      phylogenies.\n",
            "\n",
            "  \u001b[34mgneiss\u001b[0m              Plugin for building\n",
            "                      compositional models.\n",
            "\n",
            "  \u001b[34mlongitudinal\u001b[0m        Plugin for paired sample and\n",
            "                      time series analyses.\n",
            "\n",
            "  \u001b[34mmetadata\u001b[0m            Plugin for working with\n",
            "                      Metadata.\n",
            "\n",
            "  \u001b[34mphylogeny\u001b[0m           Plugin for generating and\n",
            "                      manipulating phylogenies.\n",
            "\n",
            "  \u001b[34mquality-control\u001b[0m     Plugin for quality control\n",
            "                      of feature and sequence\n",
            "                      data.\n",
            "\n",
            "  \u001b[34mquality-filter\u001b[0m      Plugin for PHRED-based\n",
            "                      filtering and trimming.\n",
            "\n",
            "  \u001b[34msample-classifier\u001b[0m   Plugin for machine learning\n",
            "                      prediction of sample\n",
            "                      metadata.\n",
            "\n",
            "  \u001b[34mtaxa\u001b[0m                Plugin for working with\n",
            "                      feature taxonomy\n",
            "                      annotations.\n",
            "\n",
            "  \u001b[34mvsearch\u001b[0m             Plugin for clustering and\n",
            "                      dereplicating with vsearch.\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "lBA_AFHJXuau"
      },
      "source": [
        "#Import the files for each study in Google Drive"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nsSk7EqrxeU0",
        "outputId": "92283223-d904-4d1c-b7c3-f365237cf2ce"
      },
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Mounted at /content/drive\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "4MQlfMDxX08J"
      },
      "source": [
        "!cp /content/sample_data/study1/*.gz /content/drive/MyDrive/BeeTeam/studyx/ "
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UErbs9mwTna-",
        "outputId": "ab65236a-73df-4e97-a3a1-3eec18f02e98"
      },
      "source": [
        "cd /content/sample_data/"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/sample_data\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "49LH8_ObzHzK"
      },
      "source": [
        "#**Paired-end.**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "cF-keGhzH4oK"
      },
      "source": [
        "#Create a manifest for the data"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "CNZyH96d-X8V"
      },
      "source": [
        "!echo -e \"sample-id\\tforward-absolute-filepath\\treverse-absolute-filepath\" > manifest.tsv"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "yZdaIMj3vN8D"
      },
      "source": [
        "#Remove underscores from file name"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EjuPTE24S1Jk",
        "outputId": "e73dcab6-c57c-4082-c8d9-ae952291b690"
      },
      "source": [
        "%%bash\n",
        "for file in *; do\n",
        "    if [ -f \"$file\" ]; then\n",
        "        mv \"$file\" `echo \"$file\" | sed 's*_*-*g'`\n",
        "    fi\n",
        "done"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "mv: 'anscombe.json' and 'anscombe.json' are the same file\n",
            "mv: 'manifest.tsv' and 'manifest.tsv' are the same file\n",
            "mv: 'README.md' and 'README.md' are the same file\n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "qWX9D8FGkQ69",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "597e266b-1dd9-489c-d06e-04d29d988181"
      },
      "source": [
        "cd /content/sample_data/"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/sample_data\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "mlvbszQBoEWx"
      },
      "source": [
        "rm *.gz"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QZXl-pQf_YT7",
        "outputId": "37f299ab-bf11-4f72-83cc-dfb3219eaf55"
      },
      "source": [
        "ls *.gz"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "A1-S1-L001-R1-001.fastq.gz   D8-S35-L001-R1-001.fastq.gz\n",
            "A1-S1-L001-R2-001.fastq.gz   D8-S35-L001-R2-001.fastq.gz\n",
            "A2-S2-L001-R1-001.fastq.gz   D9-S36-L001-R1-001.fastq.gz\n",
            "A2-S2-L001-R2-001.fastq.gz   D9-S36-L001-R2-001.fastq.gz\n",
            "A3-S3-L001-R1-001.fastq.gz   E1-S37-L001-R1-001.fastq.gz\n",
            "A3-S3-L001-R2-001.fastq.gz   E1-S37-L001-R2-001.fastq.gz\n",
            "A4-S4-L001-R1-001.fastq.gz   E2-S38-L001-R1-001.fastq.gz\n",
            "A4-S4-L001-R2-001.fastq.gz   E2-S38-L001-R2-001.fastq.gz\n",
            "A5-S5-L001-R1-001.fastq.gz   E3-S39-L001-R1-001.fastq.gz\n",
            "A5-S5-L001-R2-001.fastq.gz   E3-S39-L001-R2-001.fastq.gz\n",
            "A6-S6-L001-R1-001.fastq.gz   E4-S40-L001-R1-001.fastq.gz\n",
            "A6-S6-L001-R2-001.fastq.gz   E4-S40-L001-R2-001.fastq.gz\n",
            "A7-S7-L001-R1-001.fastq.gz   E5-S41-L001-R1-001.fastq.gz\n",
            "A7-S7-L001-R2-001.fastq.gz   E5-S41-L001-R2-001.fastq.gz\n",
            "A8-S8-L001-R1-001.fastq.gz   E6-S42-L001-R1-001.fastq.gz\n",
            "A8-S8-L001-R2-001.fastq.gz   E6-S42-L001-R2-001.fastq.gz\n",
            "A9-S9-L001-R1-001.fastq.gz   E7-S43-L001-R1-001.fastq.gz\n",
            "A9-S9-L001-R2-001.fastq.gz   E7-S43-L001-R2-001.fastq.gz\n",
            "B1-S10-L001-R1-001.fastq.gz  E8-S44-L001-R1-001.fastq.gz\n",
            "B1-S10-L001-R2-001.fastq.gz  E8-S44-L001-R2-001.fastq.gz\n",
            "B2-S11-L001-R1-001.fastq.gz  F1-S45-L001-R1-001.fastq.gz\n",
            "B2-S11-L001-R2-001.fastq.gz  F1-S45-L001-R2-001.fastq.gz\n",
            "B3-S12-L001-R1-001.fastq.gz  F2-S46-L001-R1-001.fastq.gz\n",
            "B3-S12-L001-R2-001.fastq.gz  F2-S46-L001-R2-001.fastq.gz\n",
            "B4-S13-L001-R1-001.fastq.gz  F3-S47-L001-R1-001.fastq.gz\n",
            "B4-S13-L001-R2-001.fastq.gz  F3-S47-L001-R2-001.fastq.gz\n",
            "B5-S14-L001-R1-001.fastq.gz  F4-S48-L001-R1-001.fastq.gz\n",
            "B5-S14-L001-R2-001.fastq.gz  F4-S48-L001-R2-001.fastq.gz\n",
            "B6-S15-L001-R1-001.fastq.gz  F5-S49-L001-R1-001.fastq.gz\n",
            "B6-S15-L001-R2-001.fastq.gz  F5-S49-L001-R2-001.fastq.gz\n",
            "B7-S16-L001-R1-001.fastq.gz  F6-S50-L001-R1-001.fastq.gz\n",
            "B7-S16-L001-R2-001.fastq.gz  F6-S50-L001-R2-001.fastq.gz\n",
            "B8-S17-L001-R1-001.fastq.gz  F7-S51-L001-R1-001.fastq.gz\n",
            "B8-S17-L001-R2-001.fastq.gz  F7-S51-L001-R2-001.fastq.gz\n",
            "B9-S18-L001-R1-001.fastq.gz  F8-S52-L001-R1-001.fastq.gz\n",
            "B9-S18-L001-R2-001.fastq.gz  F8-S52-L001-R2-001.fastq.gz\n",
            "C1-S19-L001-R1-001.fastq.gz  G1-S53-L001-R1-001.fastq.gz\n",
            "C1-S19-L001-R2-001.fastq.gz  G1-S53-L001-R2-001.fastq.gz\n",
            "C2-S20-L001-R1-001.fastq.gz  G2-S54-L001-R1-001.fastq.gz\n",
            "C2-S20-L001-R2-001.fastq.gz  G2-S54-L001-R2-001.fastq.gz\n",
            "C3-S21-L001-R1-001.fastq.gz  G3-S55-L001-R1-001.fastq.gz\n",
            "C3-S21-L001-R2-001.fastq.gz  G3-S55-L001-R2-001.fastq.gz\n",
            "C4-S22-L001-R1-001.fastq.gz  G4-S56-L001-R1-001.fastq.gz\n",
            "C4-S22-L001-R2-001.fastq.gz  G4-S56-L001-R2-001.fastq.gz\n",
            "C5-S23-L001-R1-001.fastq.gz  G5-S57-L001-R1-001.fastq.gz\n",
            "C5-S23-L001-R2-001.fastq.gz  G5-S57-L001-R2-001.fastq.gz\n",
            "C6-S24-L001-R1-001.fastq.gz  G6-S58-L001-R1-001.fastq.gz\n",
            "C6-S24-L001-R2-001.fastq.gz  G6-S58-L001-R2-001.fastq.gz\n",
            "C7-S25-L001-R1-001.fastq.gz  G7-S59-L001-R1-001.fastq.gz\n",
            "C7-S25-L001-R2-001.fastq.gz  G7-S59-L001-R2-001.fastq.gz\n",
            "C8-S26-L001-R1-001.fastq.gz  G8-S60-L001-R1-001.fastq.gz\n",
            "C8-S26-L001-R2-001.fastq.gz  G8-S60-L001-R2-001.fastq.gz\n",
            "C9-S27-L001-R1-001.fastq.gz  H1-S61-L001-R1-001.fastq.gz\n",
            "C9-S27-L001-R2-001.fastq.gz  H1-S61-L001-R2-001.fastq.gz\n",
            "D1-S28-L001-R1-001.fastq.gz  H2-S62-L001-R1-001.fastq.gz\n",
            "D1-S28-L001-R2-001.fastq.gz  H2-S62-L001-R2-001.fastq.gz\n",
            "D2-S29-L001-R1-001.fastq.gz  H3-S63-L001-R1-001.fastq.gz\n",
            "D2-S29-L001-R2-001.fastq.gz  H3-S63-L001-R2-001.fastq.gz\n",
            "D3-S30-L001-R1-001.fastq.gz  H4-S64-L001-R1-001.fastq.gz\n",
            "D3-S30-L001-R2-001.fastq.gz  H4-S64-L001-R2-001.fastq.gz\n",
            "D4-S31-L001-R1-001.fastq.gz  H5-S65-L001-R1-001.fastq.gz\n",
            "D4-S31-L001-R2-001.fastq.gz  H5-S65-L001-R2-001.fastq.gz\n",
            "D5-S32-L001-R1-001.fastq.gz  H6-S66-L001-R1-001.fastq.gz\n",
            "D5-S32-L001-R2-001.fastq.gz  H6-S66-L001-R2-001.fastq.gz\n",
            "D6-S33-L001-R1-001.fastq.gz  H7-S67-L001-R1-001.fastq.gz\n",
            "D6-S33-L001-R2-001.fastq.gz  H7-S67-L001-R2-001.fastq.gz\n",
            "D7-S34-L001-R1-001.fastq.gz  H8-S68-L001-R1-001.fastq.gz\n",
            "D7-S34-L001-R2-001.fastq.gz  H8-S68-L001-R2-001.fastq.gz\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "4sb4RORr-fcG"
      },
      "source": [
        "%%bash\n",
        "for f in `ls *fastq.gz | cut -d '_' -f 1 | sort -u`; do n=`basename $f`; echo -e \"${n}\\t$PWD/$f\"_1.fastq.gz \"\\t$PWD/$f\"_2.fastq.gz\" \"; done >> manifest.tsv "
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "U3yCpFk0CUlX"
      },
      "source": [
        "#Import data to Qiime2"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VkRWOEVG-oFP",
        "outputId": "b2cbe6cf-13f1-46ca-fa05-f5f4a71ed796"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime tools import \\\n",
        "  --input-path manifest.tsv \\\n",
        "  --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
        "  --input-format PairedEndFastqManifestPhred33V2 \\\n",
        "  --output-path pe-demux.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mImported manifest.tsv as PairedEndFastqManifestPhred33V2 to pe-demux.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "L0ok57E39ts5",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f6782231-dc96-4682-cc0c-0aecb02f605d"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime vsearch join-pairs \\\n",
        "  --i-demultiplexed-seqs pe-demux.qza \\\n",
        "  --o-joined-sequences demux-joined.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved SampleData[JoinedSequencesWithQuality] to: demux-joined.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "aBdoHG2tSEI9"
      },
      "source": [
        "cat /tmp/qiime2-q2cli-err-5vb3qfc9.log\n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "IDoMH4cH3i49"
      },
      "source": [
        "#DOWNLOAD demux-joined.qzv UPLOAD ON QIIME VIEW https://view.qiime2.org/"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uqhaHa9Y-oUn",
        "outputId": "2973e998-b4f6-4ebd-d4b1-06ec141756d0"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime demux summarize \\\n",
        "    --i-data  demux-joined.qza \\\n",
        "    --o-visualization demux-joined.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: demux-joined.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5WyjQ-vh-oe6",
        "outputId": "2f2889ea-3b95-4a82-d5fe-c59fe542ff8d"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime quality-filter q-score \\\n",
        "    --i-demux demux-joined.qza \\\n",
        "    --o-filtered-sequences demux-filtered.qza \\\n",
        "    --o-filter-stats demux-filter-stats.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved SampleData[JoinedSequencesWithQuality] to: demux-filtered.qza\u001b[0m\n",
            "\u001b[32mSaved QualityFilterStats to: demux-filter-stats.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "aefJ1Lb404x3"
      },
      "source": [
        ""
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "giAEy2-ErIQ8",
        "outputId": "b40f018d-b930-4226-b591-93f040140b45"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime demux summarize \\\n",
        "    --i-data pe-demux.qza \\\n",
        "    --o-visualization pe-dumux.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: pe-dumux.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Aksz8VtxoPEC",
        "outputId": "36588d9b-23bf-49ab-9c3e-09c3f1ae3748"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime dada2 denoise-paired \\\n",
        "  --i-demultiplexed-seqs pe-demux.qza \\\n",
        "  --p-trunc-len-f 200\\\n",
        "  --p-trunc-len-r 180 \\\n",
        "  --o-representative-sequences rep-seqs-dada2.qza \\\n",
        "  --o-table table-dada2.qza \\\n",
        "  --o-denoising-stats stats-dada2.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureTable[Frequency] to: table-dada2.qza\u001b[0m\n",
            "\u001b[32mSaved FeatureData[Sequence] to: rep-seqs-dada2.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[DADA2Stats] to: stats-dada2.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "3Zn8gWcnkugL"
      },
      "source": [
        "!cp /content/sample_data/*.qz* /content/drive/MyDrive/BeeTeam/study2/"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JOlmSsY6-ozc",
        "outputId": "bcb83571-4b5a-4cb4-b370-ffbce027fb1e"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime metadata tabulate \\\n",
        "   --m-input-file stats-dada2.qza \\\n",
        "   --o-visualization stats-dada2.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: stats-dada2.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "vUVbVXuW-o-B"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime feature-table tabulate-seqs \\\n",
        "    --i-data rep-seqs-deblur.qza \\\n",
        "    --o-visualization rep-seqs-deblur.qzv"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "M3SkcsTgQa8T"
      },
      "source": [
        "#Phylogenetics and Taxonomy"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ijqwFu_ww6xC"
      },
      "source": [
        "#Download table-deblur.qva and view on qiime2 https://view.qiime2.org/"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-NGA2oeTWtq0"
      },
      "source": [
        "#**2. START AGAIN HERE!**"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fB2ocVatG6FI",
        "outputId": "219e0e2d-f39d-4bb8-819d-28eb8c235e7d"
      },
      "source": [
        "cd /content/sample_data/studyx2/"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/sample_data/studyx2\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "eQG1M1WT-_VR",
        "outputId": "2e6de974-d25f-4ebb-99d0-c6ac604467d7"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime feature-table summarize \\\n",
        "    --i-table table-deblur.qza \\\n",
        "    --m-sample-metadata-file metadata.txt \\\n",
        "    --o-visualization table-deblur.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: table-deblur.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kBRvSpHjLuTD",
        "outputId": "331057c7-d475-4610-fe45-eebdf0e02da0"
      },
      "source": [
        "\n",
        "!wget -O \"sepp-refs-gg-13-8.qza\" \\\n",
        "    \"https://data.qiime2.org/2019.10/common/sepp-refs-gg-13-8.qza\""
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "--2021-02-16 14:32:57--  https://data.qiime2.org/2019.10/common/sepp-refs-gg-13-8.qza\n",
            "Resolving data.qiime2.org (data.qiime2.org)... 52.35.38.247\n",
            "Connecting to data.qiime2.org (data.qiime2.org)|52.35.38.247|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 FOUND\n",
            "Location: https://s3-us-west-2.amazonaws.com/qiime2-data/2019.10/common/sepp-refs-gg-13-8.qza [following]\n",
            "--2021-02-16 14:32:58--  https://s3-us-west-2.amazonaws.com/qiime2-data/2019.10/common/sepp-refs-gg-13-8.qza\n",
            "Resolving s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)... 52.218.196.240\n",
            "Connecting to s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)|52.218.196.240|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 50161069 (48M) [binary/octet-stream]\n",
            "Saving to: ‘sepp-refs-gg-13-8.qza’\n",
            "\n",
            "sepp-refs-gg-13-8.q 100%[===================>]  47.84M  56.0MB/s    in 0.9s    \n",
            "\n",
            "2021-02-16 14:32:59 (56.0 MB/s) - ‘sepp-refs-gg-13-8.qza’ saved [50161069/50161069]\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "-JgHT4KQvCoF",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "756e5897-658c-4f4b-c029-0af151d1d0a2"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime phylogeny align-to-tree-mafft-fasttree \\\n",
        "  --i-sequences rep-seqs-deblur.qza  \\\n",
        "  --o-alignment aligned-rep-seqs.qza \\\n",
        "  --o-masked-alignment masked-aligned-rep-seqs.qza \\\n",
        "  --o-tree unrooted-tree.qza \\\n",
        "  --o-rooted-tree rooted-tree.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/bin/bash: activate: No such file or directory\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vURaMEOyjqWi"
      },
      "source": [
        "## max-depth is taken from table-deblur.**qzv**"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "p5QNgG8ol_lB",
        "outputId": "ccbdaf25-4a6c-4d58-f113-72e2f341870d"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime diversity alpha-rarefaction \\\n",
        "  --i-table table-deblur.qza \\\n",
        "  --i-phylogeny rooted-tree.qza \\\n",
        "  --p-max-depth 1300 \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --o-visualization alpha-rarefaction.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: alpha-rarefaction.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Ml-JKqSlel3m"
      },
      "source": [
        "#sampling depth is taken from alpha-rarefaction.qzv"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jINUngswYyVI",
        "outputId": "5a3c60c4-2322-4055-a382-e73111e41bcf"
      },
      "source": [
        "#Core metrics\n",
        "!source activate qiime2-2020.11 && qiime diversity core-metrics-phylogenetic \\\n",
        "  --i-phylogeny rooted-tree.qza \\\n",
        "  --i-table table-deblur.qza \\\n",
        "  --p-sampling-depth 1300 \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --output-dir core-metrics-results"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureTable[Frequency] to: core-metrics-results/rarefied_table.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/faith_pd_vector.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/observed_features_vector.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/shannon_vector.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/evenness_vector.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/jaccard_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/bray_curtis_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/jaccard_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/bray_curtis_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/unweighted_unifrac_emperor.qzv\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/weighted_unifrac_emperor.qzv\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/jaccard_emperor.qzv\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/bray_curtis_emperor.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8wF6TKBVKeLp"
      },
      "source": [
        "alpha diversity "
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "6egchtkKY25w"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime diversity alpha-group-significance \\\n",
        "  --i-alpha-diversity core-metrics-results/faith_pd_vector.qza \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --o-visualization core-metrics-results/faith-pd-group-significance.qzv\n",
        "\n",
        "!source activate qiime2-2020.11 && qiime diversity alpha-group-significance \\\n",
        "  --i-alpha-diversity core-metrics-results/evenness_vector.qza \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --o-visualization core-metrics-results/evenness-group-significance.qzv"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Qo-tHI1FK4mB"
      },
      "source": [
        "Beta diversity "
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "OU6JjnFvVZAv"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime diversity beta-group-significance \\\n",
        "  --i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --m-metadata-column Title \\\n",
        "  --o-visualization core-metrics-results/unweighted-unifrac-group-significance.qzv \\\n",
        "  --p-pairwise"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "K1tQ3JHc7pZu"
      },
      "source": [
        "Taxonomy"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0JEQm-Kx-GYD",
        "outputId": "362d4100-3815-441b-b187-0a940c934981"
      },
      "source": [
        "!wget \\\n",
        "  -O \"gg-13-8-99-nb-classifier.qza\" \\\n",
        "  \"https://data.qiime2.org/2020.11/common/gg-13-8-99-nb-classifier.qza\""
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "--2021-01-06 16:40:51--  https://data.qiime2.org/2020.11/common/gg-13-8-99-nb-classifier.qza\n",
            "Resolving data.qiime2.org (data.qiime2.org)... 52.35.38.247\n",
            "Connecting to data.qiime2.org (data.qiime2.org)|52.35.38.247|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 FOUND\n",
            "Location: https://s3-us-west-2.amazonaws.com/qiime2-data/2020.11/common/gg-13-8-99-nb-classifier.qza [following]\n",
            "--2021-01-06 16:40:51--  https://s3-us-west-2.amazonaws.com/qiime2-data/2020.11/common/gg-13-8-99-nb-classifier.qza\n",
            "Resolving s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)... 52.218.201.216\n",
            "Connecting to s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)|52.218.201.216|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 104510980 (100M) [binary/octet-stream]\n",
            "Saving to: ‘gg-13-8-99-nb-classifier.qza’\n",
            "\n",
            "gg-13-8-99-nb-class 100%[===================>]  99.67M  30.7MB/s    in 3.2s    \n",
            "\n",
            "2021-01-06 16:40:55 (30.7 MB/s) - ‘gg-13-8-99-nb-classifier.qza’ saved [104510980/104510980]\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VkWA8v-fA5TH",
        "outputId": "fea69e23-e35f-4401-e6a6-1dc0464be8f9"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime feature-classifier classify-sklearn \\\n",
        "  --i-reads rep-seqs-deblur.qza \\\n",
        "  --i-classifier gg-13-8-99-nb-classifier.qza \\\n",
        "  --o-classification taxonomy.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureData[Taxonomy] to: taxonomy.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "E9_LmA66ZHsD",
        "outputId": "a67756cb-1d9f-4ab5-dfd7-4b3745322ce3"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime metadata tabulate \\\n",
        "  --m-input-file taxonomy.qza \\\n",
        "  --o-visualization taxonomy.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: taxonomy.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "J5YHWrkDQIN8"
      },
      "source": [
        "The min frequency here is based on the alpha rarefaction cure, add in at the ?"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "b4-XPvqiHul-"
      },
      "source": [
        "!cp -r /content/sample_data/core-results-metrics/ /content/drive/MyDrive/\n",
        "!cp  /content/sample_data/*.qz* /content/drive/MyDrive/"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "nHW5dwiCZiev"
      },
      "source": [
        "#sinlge-end"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "jF-r3YZgbFWA",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f9001b01-9145-422d-9968-ff36b49a0b62"
      },
      "source": [
        "!cp  /content/drive/MyDrive/BeeTeam/study8/*.gz /content/sample_data/"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "^C\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "0JRE8di3ZyuA"
      },
      "source": [
        "!echo -e \"sample-id\\tabsolute-filepath\" > manifest.tsv"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "gQ4xUXIYWimS"
      },
      "source": [
        ""
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "idpf6Ot9ZyuB"
      },
      "source": [
        "%%bash\n",
        "\n",
        "for f in `ls *.gz`; do n=`basename $f`; echo -e \"${n%_R1.fastq.gz}\\t$PWD/$f\"; done >> manifest.tsv\n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "l093G2LfZyuD",
        "outputId": "f979d819-b9bc-44ff-a646-5f1eea44843e"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime tools import \\\n",
        "  --input-path manifest.tsv \\\n",
        "  --type 'SampleData[SequencesWithQuality]' \\\n",
        "  --input-format SingleEndFastqManifestPhred33V2 \\\n",
        "  --output-path se-demux.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mImported manifest.tsv as SingleEndFastqManifestPhred33V2 to se-demux.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "WH5SmuumZyuE"
      },
      "source": [
        "!source activate qiime2-2020.11 &&qiime demux summarize \\\n",
        "    --i-data  se-demux.qza \\\n",
        "    --o-visualization se-demux.qzv"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_pV904uLrK_H",
        "outputId": "1fca71df-ddc9-40d9-a133-f507d56643f6"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime dada2 denoise-single \\\n",
        "  --i-demultiplexed-seqs se-demux.qza \\\n",
        "  --p-trim-left 0 \\\n",
        "  --p-trunc-len 200 \\\n",
        "  --p-n-threads 0 \\\n",
        "  --o-representative-sequences rep-seqs-dada2.qza \\\n",
        "  --o-table table-dada2.qza \\\n",
        "  --o-denoising-stats stats-dada2.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureTable[Frequency] to: table-dada2.qza\u001b[0m\n",
            "\u001b[32mSaved FeatureData[Sequence] to: rep-seqs-dada2.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[DADA2Stats] to: stats-dada2.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "bmFHOdrb6Rcx"
      },
      "source": [
        "!cp /content/sample_data/studyx2/*.qz* /content/drive/MyDrive/BeeTeam/studyx2/"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ybo4xm03uC_R",
        "outputId": "c06001a9-ab45-4a08-d98b-a46d2d48f2c9"
      },
      "source": [
        "cat /tmp/qiime2-q2cli-err-yb6hjpz2.log"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
            "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
            "\n",
            "Command: run_dada_single.R /tmp/qiime2-archive-i3pu_abe/2ab21761-5a69-4ac7-951a-2b6170b9c5ce/data /tmp/tmpd7cjviyg/output.tsv.biom /tmp/tmpd7cjviyg/track.tsv /tmp/tmpd7cjviyg 350 0 2.0 2 Inf independent consensus 1.0 1 1000000 NULL 16\n",
            "\n",
            "R version 4.0.2 (2020-06-22) \n",
            "Loading required package: Rcpp\n",
            "Warning: please export TSAN_OPTIONS='ignore_noninstrumented_modules=1' to avoid false positive reports from the OpenMP runtime!\n",
            "DADA2: 1.18.0 / Rcpp: 1.0.5 / RcppParallel: 5.0.2 \n",
            "1) Filtering The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532840_2_0_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532845_2_5_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532846_2_6_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532853_2_13_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532857_2_17_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532861_2_21_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532863_2_23_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532866_2_26_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532868_2_28_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532869_2_29_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532871_2_31_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532872_2_32_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532889_2_49_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532890_2_50_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532891_2_51_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532893_2_53_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532895_2_55_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532899_2_59_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532904_2_64_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532909_2_69_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532914_2_74_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532925_2_85_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532926_2_86_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532937_2_97_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532944_2_104_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532958_2_118_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532963_2_123_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532964_2_124_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532969_2_129_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532974_2_134_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532978_2_138_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532983_2_143_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532994_2_154_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5532997_2_157_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533006_2_166_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533010_2_170_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533011_2_171_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533014_2_174_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533017_2_177_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533018_2_178_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533020_2_180_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533023_2_183_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533027_2_187_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533028_2_188_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533030_2_190_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533036_2_196_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533040_2_200_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533045_2_205_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533047_2_207_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533048_2_208_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533051_2_211_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533052_2_212_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533053_2_213_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533055_2_215_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533057_2_217_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533062_2_222_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533066_2_226_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533073_2_233_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533083_2_243_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533097_2_257_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533100_2_260_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533104_2_264_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533105_2_265_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533106_2_266_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533109_2_269_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533111_2_271_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533116_2_276_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533119_2_279_L001_R1_001.fastq.gz not written.\n",
            "The filter removed all reads: /tmp/tmpd7cjviyg/SRR5533120_2_280_L001_R1_001.fastq.gz not written.\n",
            "Some input samples had no reads pass the filter.\n",
            "x....xx......x...x...x.x..x.xx.xx................xxx.x.x...x....x....x....x..........xx..........x......x.............x....xx....x....x...x....x..........x..x........x...xx..x..xx.x..x...xx.x.....x...x....x.xx..xxx.x.x....x...x......x.........x.............x..x...xxx..x.x....x..xx........\n",
            "2) Learning Error Rates\n",
            "202650 total bases in 579 reads from 220 samples will be used for learning the error rates.\n",
            "Error rates could not be estimated (this is usually because of very few reads).\n",
            "Error in getErrors(err, enforce = TRUE) : Error matrix is NULL.\n",
            "Execution halted\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/envs/qiime2-2020.11/lib/python3.6/site-packages/q2_dada2/_denoise.py\", line 181, in _denoise_single\n",
            "    run_commands([cmd])\n",
            "  File \"/usr/local/envs/qiime2-2020.11/lib/python3.6/site-packages/q2_dada2/_denoise.py\", line 36, in run_commands\n",
            "    subprocess.run(cmd, check=True)\n",
            "  File \"/usr/local/envs/qiime2-2020.11/lib/python3.6/subprocess.py\", line 438, in run\n",
            "    output=stdout, stderr=stderr)\n",
            "subprocess.CalledProcessError: Command '['run_dada_single.R', '/tmp/qiime2-archive-i3pu_abe/2ab21761-5a69-4ac7-951a-2b6170b9c5ce/data', '/tmp/tmpd7cjviyg/output.tsv.biom', '/tmp/tmpd7cjviyg/track.tsv', '/tmp/tmpd7cjviyg', '350', '0', '2.0', '2', 'Inf', 'independent', 'consensus', '1.0', '1', '1000000', 'NULL', '16']' returned non-zero exit status 1.\n",
            "\n",
            "During handling of the above exception, another exception occurred:\n",
            "\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/envs/qiime2-2020.11/lib/python3.6/site-packages/q2cli/commands.py\", line 329, in __call__\n",
            "    results = action(**arguments)\n",
            "  File \"<decorator-gen-516>\", line 2, in denoise_single\n",
            "  File \"/usr/local/envs/qiime2-2020.11/lib/python3.6/site-packages/qiime2/sdk/action.py\", line 245, in bound_callable\n",
            "    output_types, provenance)\n",
            "  File \"/usr/local/envs/qiime2-2020.11/lib/python3.6/site-packages/qiime2/sdk/action.py\", line 390, in _callable_executor_\n",
            "    output_views = self._callable(**view_args)\n",
            "  File \"/usr/local/envs/qiime2-2020.11/lib/python3.6/site-packages/q2_dada2/_denoise.py\", line 218, in denoise_single\n",
            "    band_size='16')\n",
            "  File \"/usr/local/envs/qiime2-2020.11/lib/python3.6/site-packages/q2_dada2/_denoise.py\", line 192, in _denoise_single\n",
            "    \" and stderr to learn more.\" % e.returncode)\n",
            "Exception: An error was encountered while running DADA2 in R (return code 1), please inspect stdout and stderr to learn more.\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PCrIv7JnZyuG",
        "outputId": "97f521b5-7203-4ca2-879c-3f9538b26bb8"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime quality-filter q-score \\\n",
        "    --i-demux se-demux.qza \\\n",
        "    --o-filtered-sequences demux-filtered.qza \\\n",
        "    --o-filter-stats demux-filter-stats.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved SampleData[SequencesWithQuality] to: demux-filtered.qza\u001b[0m\n",
            "\u001b[32mSaved QualityFilterStats to: demux-filter-stats.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fhj8yEFt5uvg",
        "outputId": "7b370d47-4adc-470b-d9c1-92344cc4a1fc"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime metadata tabulate \\\n",
        "--m-input-file demux-filter-stats.qza \\\n",
        "--o-visualization demux-filter-stats.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: demux-filter-stats.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0fZHI-ejZ8pw",
        "outputId": "01ffb1fe-f09c-4e11-b250-64c26d8d3137"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime deblur denoise-16S \\\n",
        "    --i-demultiplexed-seqs demux-filtered.qza \\\n",
        "    --p-trim-length 340 \\\n",
        "    --p-sample-stats \\\n",
        "    --o-stats deblur-stats.qza \\\n",
        "    --o-representative-sequences rep-seqs-deblur.qza \\\n",
        "    --o-table table-deblur.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[31m\u001b[1mPlugin error from deblur:\n",
            "\n",
            "  No sequences passed the filter. It is possible the trim_length (340) may exceed the longest sequence, that all of the sequences are artifacts like PhiX or adapter, or that the positive reference used is not representative of the data being denoised.\n",
            "\n",
            "Debug info has been saved to /tmp/qiime2-q2cli-err-yw3qxpq0.log\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MriIHRwma4E2",
        "outputId": "2e6de974-d25f-4ebb-99d0-c6ac604467d7"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime feature-table summarize \\\n",
        "    --i-table table-deblur.qza \\\n",
        "    --m-sample-metadata-file metadata.txt \\\n",
        "    --o-visualization table-deblur.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: table-deblur.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XjfxK9AJa4FG",
        "outputId": "756e5897-658c-4f4b-c029-0af151d1d0a2"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime phylogeny align-to-tree-mafft-fasttree \\\n",
        "  --i-sequences rep-seqs-deblur.qza  \\\n",
        "  --o-alignment aligned-rep-seqs.qza \\\n",
        "  --o-masked-alignment masked-aligned-rep-seqs.qza \\\n",
        "  --o-tree unrooted-tree.qza \\\n",
        "  --o-rooted-tree rooted-tree.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/bin/bash: activate: No such file or directory\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3hZS_xF_a4FG",
        "outputId": "ccbdaf25-4a6c-4d58-f113-72e2f341870d"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime diversity alpha-rarefaction \\\n",
        "  --i-table table-deblur.qza \\\n",
        "  --i-phylogeny rooted-tree.qza \\\n",
        "  --p-max-depth 1300 \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --o-visualization alpha-rarefaction.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: alpha-rarefaction.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gu4gTRRia4FG",
        "outputId": "5a3c60c4-2322-4055-a382-e73111e41bcf"
      },
      "source": [
        "#Core metrics\n",
        "!source activate qiime2-2020.11 && qiime diversity core-metrics-phylogenetic \\\n",
        "  --i-phylogeny rooted-tree.qza \\\n",
        "  --i-table table-deblur.qza \\\n",
        "  --p-sampling-depth 1300 \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --output-dir core-metrics-results"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureTable[Frequency] to: core-metrics-results/rarefied_table.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/faith_pd_vector.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/observed_features_vector.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/shannon_vector.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/evenness_vector.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/jaccard_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/bray_curtis_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/jaccard_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/bray_curtis_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/unweighted_unifrac_emperor.qzv\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/weighted_unifrac_emperor.qzv\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/jaccard_emperor.qzv\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/bray_curtis_emperor.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "OF7FwEQPa4FH"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime diversity alpha-group-significance \\\n",
        "  --i-alpha-diversity core-metrics-results/faith_pd_vector.qza \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --o-visualization core-metrics-results/faith-pd-group-significance.qzv\n",
        "\n",
        "!source activate qiime2-2020.11 && qiime diversity alpha-group-significance \\\n",
        "  --i-alpha-diversity core-metrics-results/evenness_vector.qza \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --o-visualization core-metrics-results/evenness-group-significance.qzv"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "_twTsOJua4FH"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime diversity beta-group-significance \\\n",
        "  --i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \\\n",
        "  --m-metadata-file metadata.txt \\\n",
        "  --m-metadata-column Title \\\n",
        "  --o-visualization core-metrics-results/unweighted-unifrac-group-significance.qzv \\\n",
        "  --p-pairwise"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PFhAioBOa4FH",
        "outputId": "362d4100-3815-441b-b187-0a940c934981"
      },
      "source": [
        "!wget \\\n",
        "  -O \"gg-13-8-99-nb-classifier.qza\" \\\n",
        "  \"https://data.qiime2.org/2020.11/common/gg-13-8-99-nb-classifier.qza\""
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "--2021-01-06 16:40:51--  https://data.qiime2.org/2020.11/common/gg-13-8-99-nb-classifier.qza\n",
            "Resolving data.qiime2.org (data.qiime2.org)... 52.35.38.247\n",
            "Connecting to data.qiime2.org (data.qiime2.org)|52.35.38.247|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 FOUND\n",
            "Location: https://s3-us-west-2.amazonaws.com/qiime2-data/2020.11/common/gg-13-8-99-nb-classifier.qza [following]\n",
            "--2021-01-06 16:40:51--  https://s3-us-west-2.amazonaws.com/qiime2-data/2020.11/common/gg-13-8-99-nb-classifier.qza\n",
            "Resolving s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)... 52.218.201.216\n",
            "Connecting to s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)|52.218.201.216|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 104510980 (100M) [binary/octet-stream]\n",
            "Saving to: ‘gg-13-8-99-nb-classifier.qza’\n",
            "\n",
            "gg-13-8-99-nb-class 100%[===================>]  99.67M  30.7MB/s    in 3.2s    \n",
            "\n",
            "2021-01-06 16:40:55 (30.7 MB/s) - ‘gg-13-8-99-nb-classifier.qza’ saved [104510980/104510980]\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6So6jqSsa4FI",
        "outputId": "fea69e23-e35f-4401-e6a6-1dc0464be8f9"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime feature-classifier classify-sklearn \\\n",
        "  --i-reads rep-seqs-deblur.qza \\\n",
        "  --i-classifier gg-13-8-99-nb-classifier.qza \\\n",
        "  --o-classification taxonomy.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureData[Taxonomy] to: taxonomy.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "r1KLFCT_a4FI",
        "outputId": "a67756cb-1d9f-4ab5-dfd7-4b3745322ce3"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime metadata tabulate \\\n",
        "  --m-input-file taxonomy.qza \\\n",
        "  --o-visualization taxonomy.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: taxonomy.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "a53KcNx4a4FI"
      },
      "source": [
        "!cp -r /content/sample_data/core-results-metrics/ /content/drive/MyDrive/\n",
        "!cp  /content/sample_data/*.qz* /content/drive/MyDrive/"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hTUfcNXvgthF",
        "outputId": "ff0da306-81a6-458c-d529-44f5eab3021b"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime tools import \\\n",
        "  --type MultiplexedSingleEndBarcodeInSequence \\\n",
        "  --input-path SRR1653719.fastq.gz \\\n",
        "  --output-path multiplexed-seqs.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mImported SRR1653719.fastq.gz as MultiplexedSingleEndBarcodeInSequenceDirFmt to multiplexed-seqs.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YxJ1g2v1jnNf",
        "outputId": "58eaac5f-146b-48d6-84f6-a5c7c4c4c644"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime tools import \\\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Usage: \u001b[34mqiime tools import\u001b[0m [OPTIONS]\n",
            "\n",
            "  Import data to create a new QIIME 2 Artifact.\n",
            "  See https://docs.qiime2.org/ for usage\n",
            "  examples and details on the file types and\n",
            "  associated semantic types that can be\n",
            "  imported.\n",
            "\n",
            "\u001b[1mOptions\u001b[0m:\n",
            "  \u001b[34m\u001b[4m--type\u001b[0m TEXT             The semantic type of\n",
            "                          the artifact that will\n",
            "                          be created upon\n",
            "                          importing. Use\n",
            "                          --show-importable-types\n",
            "                          to see what importable\n",
            "                          semantic types are\n",
            "                          available in the current\n",
            "                          deployment.   \u001b[35m[required]\u001b[0m\n",
            "  \u001b[34m\u001b[4m--input-path\u001b[0m PATH       Path to file or\n",
            "                          directory that should be\n",
            "                          imported.     \u001b[35m[required]\u001b[0m\n",
            "  \u001b[34m\u001b[4m--output-path\u001b[0m ARTIFACT  Path where output\n",
            "                          artifact should be\n",
            "                          written.      \u001b[35m[required]\u001b[0m\n",
            "  \u001b[34m--input-format\u001b[0m TEXT     The format of the data\n",
            "                          to be imported. If not\n",
            "                          provided, data must be\n",
            "                          in the format expected\n",
            "                          by the semantic type\n",
            "                          provided via --type.\n",
            "  \u001b[34m--show-importable-types\u001b[0m Show the semantic types\n",
            "                          that can be supplied to\n",
            "                          --type to import data\n",
            "                          into an artifact.\n",
            "  \u001b[34m--show-importable-formats\u001b[0m\n",
            "                          Show formats that can\n",
            "                          be supplied to\n",
            "                          --input-format to import\n",
            "                          data into an artifact.\n",
            "  \u001b[34m--help\u001b[0m                  Show this message and\n",
            "                          exit.\n",
            "\n",
            "\u001b[33m                  There were some problems with the command:                  \u001b[0m\n",
            "\u001b[31m\u001b[1m (1/3) Missing option '--type'.\u001b[0m\n",
            "\u001b[31m\u001b[1m (2/3) Missing option '--input-path'.\u001b[0m\n",
            "\u001b[31m\u001b[1m (3/3) Missing option '--output-path'.\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Ji6HbTFzO307"
      },
      "source": [
        "!gsutil -m cp /content/drive/MyDrive/Bee"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "B48WZbjGQkKB"
      },
      "source": [
        ""
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hvGz6o0MQkRp",
        "outputId": "0a2aebcf-88a5-4fe5-dac4-25854e2100bc"
      },
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Mounted at /content/drive\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "t3qwfMJYQkRp"
      },
      "source": [
        "!cp  /content/drive/MyDrive/BeeTeam/study2/table-dada2_1.qza /content/sample_data/\n",
        "!cp  /content/drive/MyDrive/BeeTeam/study3/table-dada2_2.qza /content/sample_data/\n",
        "!cp  /content/drive/MyDrive/BeeTeam/study8/table-dada2_3.qza /content/sample_data/\n",
        "!cp  /content/drive/MyDrive/BeeTeam/studyx/table-dada2_4.qza /content/sample_data/\n",
        "!cp  /content/drive/MyDrive/BeeTeam/studyx2/table-dada2_5.qza /content/sample_data/\n",
        "!cp  /content/drive/MyDrive/BeeTeam/study2/rep-seqs-dada2_1.qza /content/sample_data/\n",
        "!cp  /content/drive/MyDrive/BeeTeam/study3/rep-seqs-dada2_2.qza /content/sample_data/\n",
        "!cp  /content/drive/MyDrive/BeeTeam/study8/rep-seqs-dada2_3.qza /content/sample_data/\n",
        "!cp  /content/drive/MyDrive/BeeTeam/studyx/rep-seqs-dada2_4.qza /content/sample_data/\n",
        "!cp  /content/drive/MyDrive/BeeTeam/studyx2/rep-seqs-dada2_5.qza /content/sample_data/"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "hXw7FY3mQj-1"
      },
      "source": [
        ""
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "rP_d1LPdW1HC",
        "outputId": "9acff146-4bbd-4044-a93a-4038bfe88577"
      },
      "source": [
        "cd /content/sample_data/"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/sample_data\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fEDDxWq7T5Mt",
        "outputId": "0593ecc8-38a5-45cb-f794-588330b2150c"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime feature-table merge \\\n",
        "  --i-tables table-dada2_1.qza \\\n",
        "  --i-tables table-dada2_2.qza\\\n",
        "  --i-tables table-dada2_3.qza\\\n",
        "  --i-tables table-dada2_4.qza\\\n",
        "  --o-merged-table merge_table.qza\n",
        "!source activate qiime2-2020.11 && qiime feature-table merge-seqs \\\n",
        "    --i-data rep-seqs-dada2_1.qza \\\n",
        "    --i-data rep-seqs-dada2_2.qza \\\n",
        "    --i-data rep-seqs-dada2_3.qza \\\n",
        "    --i-data rep-seqs-dada2_4.qza \\\n",
        "   --o-merged-data merge_rep-seqs.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureTable[Frequency] to: merge_table.qza\u001b[0m\n",
            "\u001b[32mSaved FeatureData[Sequence] to: merge_rep-seqs.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HEpYzIzgvJ4D",
        "outputId": "7ad103d8-a971-4fc0-e0ac-86177d5417a1"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime feature-table summarize \\\n",
        "  --i-table merge_table.qza \\\n",
        "  --o-visualization table.qzv \\\n",
        "  --m-sample-metadata-file bee_meta.txt"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved Visualization to: table.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7t-SQXgmX38s",
        "outputId": "7ffcaa60-74a1-4836-b6f0-c55c4eb7f6e4"
      },
      "source": [
        "#Core metrics\n",
        "!source activate qiime2-2020.11 && qiime diversity core-metrics-phylogenetic \\\n",
        "  --i-phylogeny rooted-tree.qza \\\n",
        "  --i-table merge_table.qza \\\n",
        "  --p-sampling-depth 2000 \\\n",
        "  --m-metadata-file bee_meta.txt \\\n",
        "  --output-dir core-metrics-results"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureTable[Frequency] to: core-metrics-results/rarefied_table.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/faith_pd_vector.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/observed_features_vector.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/shannon_vector.qza\u001b[0m\n",
            "\u001b[32mSaved SampleData[AlphaDiversity] to: core-metrics-results/evenness_vector.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/jaccard_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved DistanceMatrix to: core-metrics-results/bray_curtis_distance_matrix.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/jaccard_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved PCoAResults to: core-metrics-results/bray_curtis_pcoa_results.qza\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/unweighted_unifrac_emperor.qzv\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/weighted_unifrac_emperor.qzv\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/jaccard_emperor.qzv\u001b[0m\n",
            "\u001b[32mSaved Visualization to: core-metrics-results/bray_curtis_emperor.qzv\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2rfwMB41X4Js",
        "outputId": "d956ae46-1767-4356-c2ee-8a921771c0f3"
      },
      "source": [
        "!wget \\\n",
        "  -O \"gg-13-8-99-nb-classifier.qza\" \\\n",
        "  \"https://data.qiime2.org/2020.11/common/gg-13-8-99-nb-classifier.qza\""
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "--2021-03-21 20:00:05--  https://data.qiime2.org/2020.11/common/gg-13-8-99-nb-classifier.qza\n",
            "Resolving data.qiime2.org (data.qiime2.org)... 54.200.1.12\n",
            "Connecting to data.qiime2.org (data.qiime2.org)|54.200.1.12|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 FOUND\n",
            "Location: https://s3-us-west-2.amazonaws.com/qiime2-data/2020.11/common/gg-13-8-99-nb-classifier.qza [following]\n",
            "--2021-03-21 20:00:05--  https://s3-us-west-2.amazonaws.com/qiime2-data/2020.11/common/gg-13-8-99-nb-classifier.qza\n",
            "Resolving s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)... 52.218.182.32\n",
            "Connecting to s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)|52.218.182.32|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 104510980 (100M) [binary/octet-stream]\n",
            "Saving to: ‘gg-13-8-99-nb-classifier.qza’\n",
            "\n",
            "gg-13-8-99-nb-class 100%[===================>]  99.67M  29.8MB/s    in 3.5s    \n",
            "\n",
            "2021-03-21 20:00:09 (28.3 MB/s) - ‘gg-13-8-99-nb-classifier.qza’ saved [104510980/104510980]\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-upX9LcDX4J1",
        "outputId": "45f55bd1-81d5-47db-c790-c6b6986a4a8f"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime feature-classifier classify-sklearn \\\n",
        "  --i-reads merge_rep-seqs.qza \\\n",
        "  --i-classifier gg-13-8-99-nb-classifier.qza \\\n",
        "  --o-classification taxonomy.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureData[Taxonomy] to: taxonomy.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "khygin02X4J3",
        "outputId": "f92d68d8-a6ca-44de-ec4d-b3f5e2583c7c"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime metadata tabulate \\\n",
        "  --m-input-file taxonomy.qza \\\n",
        "  --o-visualization merged_taxonomy.qzv"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[31m\u001b[1mThere was an issue with loading the file taxonomy.qza as metadata:\n",
            "\n",
            "  Metadata file path doesn't exist, or the path points to something other than a file. Please check that the path exists, has read permissions, and points to a regular file (not a directory): taxonomy.qza\n",
            "\n",
            "  There may be more errors present in the metadata file. To get a full report, sample/feature metadata files can be validated with Keemei: https://keemei.qiime2.org\n",
            "\n",
            "  Find details on QIIME 2 metadata requirements here: https://docs.qiime2.org/2020.11/tutorials/metadata/\u001b[0m\n",
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "k12XjVxpYeWt",
        "outputId": "ba5133b4-f324-4706-f9a8-1e274d06a8dd"
      },
      "source": [
        "!source activate qiime2-2020.11 && qiime phylogeny align-to-tree-mafft-fasttree \\\n",
        "  --i-sequences merge_rep-seqs.qza  \\\n",
        "  --o-alignment aligned-rep-seqs.qza \\\n",
        "  --o-masked-alignment masked-aligned-rep-seqs.qza \\\n",
        "  --o-tree unrooted-tree.qza \\\n",
        "  --o-rooted-tree rooted-tree.qza"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\u001b[32mSaved FeatureData[AlignedSequence] to: aligned-rep-seqs.qza\u001b[0m\n",
            "\u001b[32mSaved FeatureData[AlignedSequence] to: masked-aligned-rep-seqs.qza\u001b[0m\n",
            "\u001b[32mSaved Phylogeny[Unrooted] to: unrooted-tree.qza\u001b[0m\n",
            "\u001b[32mSaved Phylogeny[Rooted] to: rooted-tree.qza\u001b[0m\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "xfmpzTIfYPPb"
      },
      "source": [
        "!cp -r /content/sample_data/core-metrics-results/ /content/drive/MyDrive/BeeTeam/merge/"
      ],
      "execution_count": null,
      "outputs": []
    }
  ]
}