# Bee_Microbiome

The repository "Bee_Microbiome" to ensure availability of the data and code files used for the report: "A Meta-Analysis on the 16s-rRNA Gut Microbiome Data of Honey Bees 	(_Apis Mellifera_)". The paper was submitted to complete the BIOC0023 Level 6 module of the BSc Biochemistry course at UCL and it is currently being worked on for publication. **Therefore the repository is subject to continuous changes**. 

The data and the files uploaded here allow and instruct the user to investigate the microbiome from 4 16s-rRNA amplicon sequencing studies carried out on samples of bee microbiomes. These 4 studies were the only ones out of a body of 26 (as of the ends of October 2020) which made their data available, labelled appropriately, and provided barcodes for demultiplexing where necessary. 

## Rationale

Bees are key pollinators for the vegetation of terrestrial ecosystems and, similarly to every animal, they are associated with a microbiome which is fundamental to the preservation of the hostâ€™s health. However, except for some pathogenic species, only some members of the honey bee microbiome have been identified and fewer have been characterised. Identifying the core microbiome, such as with the 2008 Human Microbiome Project, is fundamental to truly understand population differences induced by disease states, drugs and other changes affecting the microbiome. In this paper, we carry out a meta-analysis of existing 16s-rRNA metagenomics experiments to merge the available data, analyse it, and make a first step towards a honey bee core microbiome. In our findings, we identify Snodgrassella, Lactobacillus, Bifidobacterium, Fructobacillus and Saccaribacter as the most prevalent genuses across all samples from the 4 studies available. Moreover, we compiled the honey bee rare microbiome containing more than 200 bacterial genuses which were present in only one sample of our dataset. The study was limited due the low quantity and quality of available datasets, caused by datasets with samples affected by parasitic infestation and by the general difficulty in obtaining high quantities of DNA from honey bees guts. Finally, we plan to enrich the dataset with new publications and reiterate analyses periodically to keep improving our blueprint for the core honey bee microbiome.

**VISIT THE [WIKI](https://github.com/JacopoGabrielli/Bee_Microbiome/wiki) FOR CLEAR GUIDANCE IN NAVIGATING THE REPOSITORY**

## Acknowledgments

Jacopo Gabrielli and Alexis Gkantiragas (co-authors) would like to thank Dr. Toryn Poolman for the help and advice he provided us throughout the project. Additionally, a special thanks to the other students whom we have had the chance to discuss with and who stimulated us to continuously challenge our results. 








